<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>GenesetAnalysisWorkflow • GenesetAnalysisWorkflow</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="GenesetAnalysisWorkflow">
<meta property="og:description" content="This workflow package shows how transcriptomic signatures can be used
  to infer phenotypes. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">GenesetAnalysisWorkflow</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/workflow_singscore_vissE.html">Visualising molecular phenotypes inferred from individual samples using singscore, vissE and msigdb</a>
    </li>
    <li>
      <a href="articles/workflow_singscore_vissE_Chinese_ver.html">使用R包singscore，vissE和msigdb对单样本进行分子表型可视化</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/DavisLaboratory/GenesetAnalysisWorkflow">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="gene-set-enrichment-analysis-workshop" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#gene-set-enrichment-analysis-workshop" class="anchor"></a>Gene-set enrichment analysis workshop</h1></div>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>This workshop will focus on performing gene-set enrichment analysis of transcriptomic data and visualising the results of enrichment analysis. We will perform single-sample gene-set enrichment using methods in the <code>singscore</code> package to explore molecular phenotypes in individual samples. Following this, we will perform gene-set enrichment analysis using tools from the <code>limma</code> and <code>edgeR</code> packages. Finally, we will demonstrate a graph-based approach to visualise, summarise and interpret resutls of gene-set enrichment analysis.</p>
<p>The workshop will be organised into two broad sections:</p>
<ul>
<li>Molecular phenotyping of individual samples</li>
<li>Identifying and visualising higher-order phenotypes</li>
</ul>
<p>Detailed material can be found <a href="https://davislaboratory.github.io/GenesetAnalysisWorkflow/articles/workflow_singscore_vissE.html">here</a>.</p>
</div>
<div id="pre-requisites" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h2>
<p>The course is aimed at PhD students, Master’s students, and third &amp; fourth year undergraduate students. Some basic R knowledge is assumed - this is not an introduction to R course. If you are not familiar with the R statistical programming language it is compulsory that you work through an introductory R course before you attend this workshop.</p>
</div>
<div id="r-packages-used" class="section level2">
<h2 class="hasAnchor">
<a href="#r-packages-used" class="anchor"></a><em>R</em> packages used</h2>
<p>The following key R packages will be used:</p>
<ul>
<li><code>singscore</code></li>
<li><code>vissE</code></li>
<li><code>msigdb</code></li>
<li><code>emtdata</code></li>
<li><code>edgeR</code></li>
<li><code>limma</code></li>
<li><code>GSEABase</code></li>
<li><code>igraph</code></li>
</ul>
</div>
<div id="time-outline" class="section level2">
<h2 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h2>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction &amp; setup</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Part 1. Molecular phenotyping of individual samples</td>
<td>45m</td>
</tr>
<tr class="odd">
<td>Part 2. Identifying and visualising higher-order phenotypes</td>
<td>45m</td>
</tr>
<tr class="even">
<td>Q &amp; A</td>
<td>10m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h2>
<div id="learning-goals" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h3>
<ul>
<li>Learn how to perform gene-set testing in R.</li>
<li>Understand the results of gene-set enrichment analysis.</li>
<li>Understand the importance of visualisation in bioinformatics and computational biology.</li>
</ul>
</div>
<div id="learning-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h3>
<ul>
<li>Perform a gene-set enrichment analysis and interpret the results.</li>
<li>Apply vissE to identify higher-order phenotypes and to visualise the results of any gene-set enrichment analysis.</li>
</ul>
</div>
</div>
<div id="workshop-package-installation" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-package-installation" class="anchor"></a>Workshop package installation</h2>
<div id="guide" class="section level3">
<h3 class="hasAnchor">
<a href="#guide" class="anchor"></a>Guide</h3>
<p>This is necessary in order to reproduce the code shown in the workshop. The workshop is designed for R <code>4.1</code> and can be installed using one of the two ways below.</p>
</div>
<div id="via-docker-image" class="section level3">
<h3 class="hasAnchor">
<a href="#via-docker-image" class="anchor"></a>Via Docker image</h3>
<p>If you’re familiar with <a href="https://docs.docker.com/get-docker/">Docker</a> you could use the Docker image which has all the software pre-configured to the correct versions.</p>
<pre><code>docker run -e PASSWORD=password -p 8787:8787 bhuvad/genesetanalysisworkflow:latest</code></pre>
<p>Once running, navigate to <a href="http://localhost:8787/" class="uri">http://localhost:8787/</a> and then login with <code>Username:rstudio</code> and <code>Password:password</code>.</p>
<p>You should see the Rmarkdown file with all the workshop code which you can run.</p>
</div>
<div id="via-github" class="section level3">
<h3 class="hasAnchor">
<a href="#via-github" class="anchor"></a>Via GitHub</h3>
<p>Alternatively, you could install the workshop using the commands below in R <code>4.1</code>.</p>
<pre><code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages('remotes')

# Install workshop package
remotes::install_github("DavisLaboratory/GenesetAnalysisWorkflow", build_vignettes = TRUE)

# To view vignettes
library(GenesetAnalysisWorkflow)
browseVignettes("GenesetAnalysisWorkflow")</a></code></pre>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/DavisLaboratory/GenesetAnalysisWorkflow/">https://​github.com/​DavisLaboratory/​GenesetAnalysisWorkflow/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/DavisLaboratory/GenesetAnalysisWorkflow/issues">https://​github.com/​DavisLaboratory/​GenesetAnalysisWorkflow/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Dharmesh D. Bhuva <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-6398-9157" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Chin Wee Tan <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-9695-7218" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Melissa J. Davis <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-4864-7033" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Dharmesh D. Bhuva, Chin Wee Tan, Melissa J. Davis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
