<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Visualising molecular phenotypes inferred from individual samples using singscore, vissE and msigdb • GenesetAnalysisWorkflow</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Visualising molecular phenotypes inferred from individual samples using singscore, vissE and msigdb">
<meta property="og:description" content="GenesetAnalysisWorkflow">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenesetAnalysisWorkflow</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/workflow_singscore_vissE.html">Visualising molecular phenotypes inferred from individual samples using singscore, vissE and msigdb</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/DavisLaboratory/GenesetAnalysisWorkflow">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workflow_singscore_vissE_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Visualising molecular phenotypes inferred from individual samples using singscore, vissE and msigdb</h1>
                        <h4 class="author">Dharmesh D Bhuva</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of Medical Research, Parkville, VIC 3052, AustraliaDepartment of Medical Biology, University of Melbourne, Parkville, VIC 3010, Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:bhuva.d@wehi.edu.au" class="email">bhuva.d@wehi.edu.au</a>
      </address>
                              <h4 class="author">Chin Wee Tan</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of Medical Research, Parkville, VIC 3052, AustraliaDepartment of Medical Biology, University of Melbourne, Parkville, VIC 3010, Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:cwtan@wehi.edu.au" class="email">cwtan@wehi.edu.au</a>
      </address>
                              <h4 class="author">Melissa J Davis</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of Medical Research, Parkville, VIC 3052, AustraliaDepartment of Medical Biology, University of Melbourne, Parkville, VIC 3010, AustraliaDepartment of Biochemistry and Molecular Biology, Faculty of Medicine, Dentistry and Health Sciences, University of Melbourne, Parkville, VIC, 3010, Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:davis.m@wehi.edu.au" class="email">davis.m@wehi.edu.au</a>
      </address>
                  
            <h4 class="date">Aug 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/GenesetAnalysisWorkflow/blob/master/vignettes/workflow_singscore_vissE.Rmd"><code>vignettes/workflow_singscore_vissE.Rmd</code></a></small>
      <div class="hidden name"><code>workflow_singscore_vissE.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Abstract</p>
    </div>
    
<p>
</p>
<p><strong>R version</strong>: R version 4.1.0 (2021-05-18) <br><strong>Bioconductor version</strong>: 3.14 <br><strong>Package version</strong>: 0.9.1</p>

<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>A standard bioinformatics analysis of ’omics data will produce a list of molecules following statistical analysis. In the context of transcriptomics, these molecules are genes or transcripts and the statistical approach used to identify them is mostly a differential expression analysis. Once genes have been identified as differentially expressed in an experiment, biologists are often interested in understanding their biological implications. This is done by understanding their functional role in the biological system being investigated. The role and function of many genes is known to some extent and this is an area of continued research. Knowledge on gene function is often encoded into knowledgebases such as the gene ontology and other pathway databases. Given these functional annotations, we are interested in identifying over-represented functions in our data.</p>
<p>To do so, we use gene-set enrichment analysis, a group of methods designed to identify enriched functions represented by collections of genes known as gene-sets. This workflow will demonstrate functional analysis of transcriptomic data using the molecular signatures database (through the <code>msigdb</code> R/Bioconductor package) and a gene-set enrichment method, <code>singscore</code>. It will also demonstrate how higher order biological themes can be identified in data using the <code>vissE</code>package. It will begin by loading gene expression data and gene-sets from the <code>ExperimentHub</code> using the <code>emtdata</code> and <code>msigdb</code> R/Bioconductor packages. Molecular phenotypes representing the functional characterisitic of Samples will be identified using the single-sample gene-set enrichment method, singscore. Finally, higher-order functional themes will be identified using vissE.</p>
</div>
<div id="description-of-the-biological-problem" class="section level1">
<h1 class="hasAnchor">
<a href="#description-of-the-biological-problem" class="anchor"></a>Description of the biological problem</h1>
<p>Epithelial–mesenchymal transition (EMT) is a cellular process where static and polarised epithelial cells convert to a motile mesenchymal phenotype. It is the fundamental mechanism involved in embryogenesis and tissue differentiation and in the context of cancer, implicated in the cancer progression and metastasis. EMT entails a progressive loss of epithelial characteristics and acquisition of mesenchymal features, which in cancer enables cells to acquire migratory and invasive properties, induces stemness, avoid apoptosis and senescence and become immune-resistant. [Yilmaz2009, Thiery2009, Polyak2009, Książkiewicz2012]</p>
<p>EMT is characterised by downregulation of epithelial markers (e.g E-cadherin, claudins) and upregulation of mesenchymal markers (e.g., vimentin, N-cadherin)), resulting in the loss of cell–cell adhesion and cell polarity, and the acquisition of migratory and invasive properties. There is a switch between keratin-rich adheren junction network to a vimentin-rich focal adhesion network with cadherin switch from E-cadherin to N-cadherin.[Tomita2000]</p>
<p>Although cancer cells utilise EMT to facilitating metastatic spread, they rarely fully convert to the mesenchymal phenotype, and instead presents a hybrid phenotype loosely termed epithelial mesenchymal plasticity (EMP). EMP has been associated with increased metastasis risk and poor prognosis in breast cancer <span class="citation">(Cursons et al. 2015)</span>. There are currently limited specific markers proposed for this transient/metastable phenotype, owing to the lack of reliable progression readout and consequent limited studies for the transitional and progression states of EMT. Identification of specific markers of this hybrid state will provide crucial information to identify interventions for metastasis as well as mechanistic insights to uncover potential therapeutics. [Ribeiro2015]</p>
<p>In this workflow, we utilise bulk transcriptomics data from <span class="citation">(Cursons et al. 2015)</span> study which interrogates EMP in two breast cancer cell line models, namely the PMC42 system (PMC42-ET and PMC42-LA sublines) and MDA-MB-468 cells. There are numerous avenues to induce EMT, with epidermal growth factor (EGF) signalling via the EGFRs being one such stimuli. Notably signalling via EGFRs is also well-established driver of breast cancer progression. In this study, EMT was induced using EGF stimulation, and by hypoxia for MDA-MB-468 cells.<span class="citation">(Cursons et al. 2015)</span></p>
<p>PMC42-LA is an epithelial subline derived from the vimentin positive, E-Cadherin negative parental PMC42-ET cells with both classified as having a “Basal B” transcriptome (expressed mesenchymal gene, lacked epithelial marker expression). Both of the PMC42 cell-lines undergo EGF-driven EMT, but PMC42-ET’s response is reduced due to higher basal vimentin expression.<span class="citation">(Cursons et al. 2015)</span></p>
</div>
<div id="molecular-phenotyping-of-individual-samples" class="section level1">
<h1 class="hasAnchor">
<a href="#molecular-phenotyping-of-individual-samples" class="anchor"></a>Molecular phenotyping of individual samples</h1>
<p>To preform molecular phenotyping of the data from <span class="citation">(Cursons et al. 2015)</span>, we need the RNA-seq counts and appropriate gene-sets that represent the molecular phenotypes of interest. We will use processed RNA-seq data from the <code>emtdata</code> package in this workflow. We will then score individual samples against custom gene signatures from <span class="citation">(Tan et al. 2014)</span> and from the molecular signatures database (MSigDB) (<span class="citation">Subramanian et al. (2005)</span>) using singscore (<span class="citation">Bhuva, Cursons, and Davis (2020)</span>).</p>
<div id="load-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h2>
<p>The <code>emtdata</code> package contains multiple pre-processed RNA-seq data from experiments studying the epithelial to mesenchymal transition. The package contains the following processed datasets:</p>
<ol style="list-style-type: decimal">
<li>cursons2015_se - RNA-seq data from various breast cancer cell lines representing different degrees of epithelial and mesenchymal phenotypes <span class="citation">(Cursons et al. 2015)</span>.</li>
<li>cursons2018_se - RNA-seq data from the human mammary epithelial (HMLE) cell line treated with TGF<span class="math inline">\(\beta\)</span> to induce an epithelial to mesenchymal transition (EMT) which is followed by treatment with miR-200c to induce a mesenchymal to epithelial transition (MET) <span class="citation">(Cursons et al. 2018)</span>.</li>
<li>foroutan2017_se - A compendium of EMT induction experiments combined and corrected for batch effects <span class="citation">(Foroutan et al. 2017)</span>.</li>
</ol>
<p>The data used in this workflow is the <span class="citation">(Cursons et al. 2015)</span> RNA-seq data. RNA-seq reads downloaded from the sequencing read archive (SRA) and processed using the Subread aligner <span class="citation">(Liao, Smyth, and Shi 2013)</span> and featureCounts <span class="citation">(Liao, Smyth, and Shi 2014)</span> to generate a counts matrix. Following this, genes with low expression were filtered out using <code><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html">edgeR::filterByExpr()</a></code>. Normalisation factors were computed using TMM <span class="citation">(Robinson and Oshlack 2010)</span> and reads per kilobase per million (RPKM) values computed using function in the <code>edgeR</code> R/Bioconductor package <span class="citation">(Chen et al. 2021)</span>. Data were then stored in a SummarizedExperiment object that can be retrieved from the <code>ExperimentHub</code> using the <code>emtdata</code> R/Bioconductor package.</p>
<p>Data from the <code>emtdata</code> package can be retireved by using accessors within the package or by querying the <code>ExperimentHub</code>. The query below searches for all objects in the hub associated with the search term “emtdata.” Information about three datasets available in the <code>emtdata</code> package is retrieved.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span>

<span class="va">eh</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">res</span> <span class="op">=</span> <span class="fu">query</span><span class="op">(</span><span class="va">eh</span>, <span class="st">'emtdata'</span><span class="op">)</span>
<span class="va">res</span></code></pre></div>
<pre><code>## ExperimentHub with 3 records
## # snapshotDate(): 2021-08-18
## # $dataprovider: Walter and Eliza Hall Institute of Medical Research, Queens...
## # $species: Homo sapiens
## # $rdataclass: GSEABase::SummarizedExperiment
## # additional mcols(): taxonomyid, genome, description,
## #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,
## #   rdatapath, sourceurl, sourcetype 
## # retrieve records with, e.g., 'object[["EH5439"]]' 
## 
##            title          
##   EH5439 | foroutan2017_se
##   EH5440 | cursons2018_se 
##   EH5441 | cursons2015_se</code></pre>
<p>We can view all metadata associated with each object using the <code>mcols()</code> function. This information links back to the original publication for each data. We can also see information on the species this data was generated from.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#retrieve metadata for records</span>
<span class="fu">mcols</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 3 rows and 15 columns
##                  title           dataprovider      species taxonomyid
##            &lt;character&gt;            &lt;character&gt;  &lt;character&gt;  &lt;integer&gt;
## EH5439 foroutan2017_se Walter and Eliza Hal.. Homo sapiens       9606
## EH5440  cursons2018_se Queensland Universit.. Homo sapiens       9606
## EH5441  cursons2015_se Center for Cancer Bi.. Homo sapiens       9606
##             genome            description coordinate_1_based
##        &lt;character&gt;            &lt;character&gt;          &lt;integer&gt;
## EH5439          NA Gene expression data..                  1
## EH5440          NA Gene expression data..                  1
## EH5441          NA Gene expression data..                  1
##                    maintainer rdatadateadded preparerclass
##                   &lt;character&gt;    &lt;character&gt;   &lt;character&gt;
## EH5439 Malvika D. Kharbanda..     2021-03-30       emtdata
## EH5440 Malvika D. Kharbanda..     2021-03-30       emtdata
## EH5441 Malvika D. Kharbanda..     2021-03-30       emtdata
##                                             tags             rdataclass
##                                           &lt;list&gt;            &lt;character&gt;
## EH5439         Gene_Expression,Homo_sapiens_Data GSEABase::Summarized..
## EH5440                    HMLE,Homo_sapiens_Data GSEABase::Summarized..
## EH5441 Homo_sapiens_Data,MDA-MB-468,PMC42-ET,... GSEABase::Summarized..
##                     rdatapath              sourceurl  sourcetype
##                   &lt;character&gt;            &lt;character&gt; &lt;character&gt;
## EH5439 emtdata/foroutan2017.. https://doi.org/10.4..         TXT
## EH5440 emtdata/cursons2018_.. https://www.ncbi.nlm..         TXT
## EH5441 emtdata/cursons2015_.. https://www.ncbi.nlm..         TXT</code></pre>
<p>Objects can then be retrieved using the accession ID (“EH5441”) or using the function <code><a href="https://rdrr.io/pkg/emtdata/man/cursons2015_se.html">cursons2015_se()</a></code>. Data are stored as <code>SummarizedExperiment</code> objects and the functions <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData()</a></code>, <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData()</a></code> and <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay()</a></code> can be used to interact with the object. We see that the data has measurements for approximately 30000 genes across 21 samples. Both count-level data and logRPKM measurements are stored in the object.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/DavisLaboratory/emtdata">emtdata</a></span><span class="op">)</span>

<span class="co">#load using object ID</span>
<span class="va">emt_se</span> <span class="op">=</span> <span class="va">eh</span><span class="op">[[</span><span class="st">'EH5441'</span><span class="op">]</span><span class="op">]</span>

<span class="co">#alternative approach</span>
<span class="va">emt_se</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/emtdata/man/cursons2015_se.html">cursons2015_se</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">emt_se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 29866 21 
## metadata(0):
## assays(2): counts logRPKM
## rownames(29866): ENSG00000223972 ENSG00000227232 ... ENSG00000271254
##   ENSG00000275405
## rowData names(7): Chr Start ... gene_name gene_biotype
## colnames(21): MDA468_Ctrl_Rep1 MDA468_Ctrl_Rep2 ... PMC42LA_EGF_Rep2
##   PMC42LA_EGF_Rep3
## colData names(14): group lib.size ... Organism SRA.Study</code></pre>
<p>The sample annotations show us that there are 3 cell lines in this data (PMC42-ET, PMC42-LA and MDA-MB-648). These cell lines have been treated with Hypoxia and EGF with measurements taken either 3 or 7 days post treatment. We also get metadata associated with each sample, including the original SRA identifier and their library sizes.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#view sample annotations</span>
<span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">emt_se</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##                   group  lib.size norm.factors                              Run
## MDA468_Ctrl_Rep1      1 114794500    0.9743731 SRR3576739,SRR3576768,SRR3591405
## MDA468_Ctrl_Rep2      1 114491243    0.9931838 SRR3576740,SRR3576770,SRR3591406
## MDA468_Ctrl_Rep3      1 107923317    1.0358755 SRR3576751,SRR3576784,SRR3591417
## MDA468_EGF_Rep1       1 120949415    0.9793903 SRR3576753,SRR3576786,SRR3591419
## MDA468_EGF_Rep2       1 108830760    0.9865599 SRR3576754,SRR3576788,SRR3591420
## MDA468_EGF_Rep3       1  95608850    0.9006452 SRR3576755,SRR3576790,SRR3591421
## MDA468_HPX_Rep1       1 103636210    0.9782668 SRR3576756,SRR3576791,SRR3591422
## MDA468_HPX_Rep2       1  99763188    1.0218358 SRR3576757,SRR3576792,SRR3591423
## MDA468_HPX_Rep3       1 111820864    0.9914170 SRR3576758,SRR3576793,SRR3591424
## PMC42ET_Ctrl_Rep1     1  71276286    1.0443833 SRR3576759,SRR3576795,SRR3591425
## PMC42ET_Ctrl_Rep2     1  74625943    0.9957722 SRR3576741,SRR3576771,SRR3591407
## PMC42ET_Ctrl_Rep3     1 104167668    1.0622063 SRR3576742,SRR3576772,SRR3591408
## PMC42ET_EGF_Rep1      1  74711695    1.0303307 SRR3576743,SRR3576773,SRR3591409
## PMC42ET_EGF_Rep2      1  87532741    1.0312069 SRR3576744,SRR3576775,SRR3591410
## PMC42ET_EGF_Rep3      1 102691326    1.0485896 SRR3576745,SRR3576776,SRR3591411
## PMC42LA_Ctrl_Rep1     1  66771046    0.9511911 SRR3576746,SRR3576777,SRR3591412
## PMC42LA_Ctrl_Rep2     1  89935383    0.9872186 SRR3576747,SRR3576779,SRR3591413
## PMC42LA_Ctrl_Rep3     1  74822017    0.9930820 SRR3576748,SRR3576781,SRR3591414
## PMC42LA_EGF_Rep1      1  80151428    0.9440959 SRR3576749,SRR3576782,SRR3591415
## PMC42LA_EGF_Rep2      1  88233254    1.0227524 SRR3576750,SRR3576783,SRR3591416
## PMC42LA_EGF_Rep3      1  85941564    1.0437588 SRR3576752,SRR3576785,SRR3591418
##                         Sample.Name    Subline    Treatment  BioProject
## MDA468_Ctrl_Rep1   MDA468_Ctrl_Rep1 MDA-MB-468      control PRJNA322427
## MDA468_Ctrl_Rep2   MDA468_Ctrl_Rep2 MDA-MB-468      control PRJNA322427
## MDA468_Ctrl_Rep3   MDA468_Ctrl_Rep3 MDA-MB-468      control PRJNA322427
## MDA468_EGF_Rep1     MDA468_EGF_Rep1 MDA-MB-468     EGF - 7d PRJNA322427
## MDA468_EGF_Rep2     MDA468_EGF_Rep2 MDA-MB-468     EGF - 7d PRJNA322427
## MDA468_EGF_Rep3     MDA468_EGF_Rep3 MDA-MB-468     EGF - 7d PRJNA322427
## MDA468_HPX_Rep1     MDA468_HPX_Rep1 MDA-MB-468 Hypoxia - 7d PRJNA322427
## MDA468_HPX_Rep2     MDA468_HPX_Rep2 MDA-MB-468 Hypoxia - 7d PRJNA322427
## MDA468_HPX_Rep3     MDA468_HPX_Rep3 MDA-MB-468 Hypoxia - 7d PRJNA322427
## PMC42ET_Ctrl_Rep1 PMC42ET_Ctrl_Rep1   PMC42-ET      control PRJNA322427
## PMC42ET_Ctrl_Rep2 PMC42ET_Ctrl_Rep2   PMC42-ET      control PRJNA322427
## PMC42ET_Ctrl_Rep3 PMC42ET_Ctrl_Rep3   PMC42-ET      control PRJNA322427
## PMC42ET_EGF_Rep1   PMC42ET_EGF_Rep1   PMC42-ET     EGF - 3d PRJNA322427
## PMC42ET_EGF_Rep2   PMC42ET_EGF_Rep2   PMC42-ET     EGF - 3d PRJNA322427
## PMC42ET_EGF_Rep3   PMC42ET_EGF_Rep3   PMC42-ET     EGF - 3d PRJNA322427
## PMC42LA_Ctrl_Rep1 PMC42LA_Ctrl_Rep1   PMC42-LA      control PRJNA322427
## PMC42LA_Ctrl_Rep2 PMC42LA_Ctrl_Rep2   PMC42-LA      control PRJNA322427
## PMC42LA_Ctrl_Rep3 PMC42LA_Ctrl_Rep3   PMC42-LA      control PRJNA322427
## PMC42LA_EGF_Rep1   PMC42LA_EGF_Rep1   PMC42-LA     EGF - 3d PRJNA322427
## PMC42LA_EGF_Rep2   PMC42LA_EGF_Rep2   PMC42-LA     EGF - 3d PRJNA322427
## PMC42LA_EGF_Rep3   PMC42LA_EGF_Rep3   PMC42-LA     EGF - 3d PRJNA322427
##                      BioSample                         Center.Name
## MDA468_Ctrl_Rep1  SAMN05162532 Queensland University of Technology
## MDA468_Ctrl_Rep2  SAMN05162533 Queensland University of Technology
## MDA468_Ctrl_Rep3  SAMN05162534 Queensland University of Technology
## MDA468_EGF_Rep1   SAMN05162535 Queensland University of Technology
## MDA468_EGF_Rep2   SAMN05162536 Queensland University of Technology
## MDA468_EGF_Rep3   SAMN05162537 Queensland University of Technology
## MDA468_HPX_Rep1   SAMN05162538 Queensland University of Technology
## MDA468_HPX_Rep2   SAMN05162539 Queensland University of Technology
## MDA468_HPX_Rep3   SAMN05162540 Queensland University of Technology
## PMC42ET_Ctrl_Rep1 SAMN05162541 Queensland University of Technology
## PMC42ET_Ctrl_Rep2 SAMN05162542 Queensland University of Technology
## PMC42ET_Ctrl_Rep3 SAMN05162543 Queensland University of Technology
## PMC42ET_EGF_Rep1  SAMN05162544 Queensland University of Technology
## PMC42ET_EGF_Rep2  SAMN05162545 Queensland University of Technology
## PMC42ET_EGF_Rep3  SAMN05162546 Queensland University of Technology
## PMC42LA_Ctrl_Rep1 SAMN05162547 Queensland University of Technology
## PMC42LA_Ctrl_Rep2 SAMN05162548 Queensland University of Technology
## PMC42LA_Ctrl_Rep3 SAMN05162549 Queensland University of Technology
## PMC42LA_EGF_Rep1  SAMN05162550 Queensland University of Technology
## PMC42LA_EGF_Rep2  SAMN05162551 Queensland University of Technology
## PMC42LA_EGF_Rep3  SAMN05162552 Queensland University of Technology
##                                         Experiment  Cell.Line     Organism
## MDA468_Ctrl_Rep1  SRX1795037,SRX1795064,SRX1802009 MDA-MB-468 Homo sapiens
## MDA468_Ctrl_Rep2  SRX1795038,SRX1795066,SRX1802010 MDA-MB-468 Homo sapiens
## MDA468_Ctrl_Rep3  SRX1795049,SRX1795078,SRX1802021 MDA-MB-468 Homo sapiens
## MDA468_EGF_Rep1   SRX1795051,SRX1795080,SRX1802023 MDA-MB-468 Homo sapiens
## MDA468_EGF_Rep2   SRX1795052,SRX1795081,SRX1802024 MDA-MB-468 Homo sapiens
## MDA468_EGF_Rep3   SRX1795053,SRX1795082,SRX1802025 MDA-MB-468 Homo sapiens
## MDA468_HPX_Rep1   SRX1795054,SRX1795083,SRX1802026 MDA-MB-468 Homo sapiens
## MDA468_HPX_Rep2   SRX1795055,SRX1795084,SRX1802027 MDA-MB-468 Homo sapiens
## MDA468_HPX_Rep3   SRX1795056,SRX1795086,SRX1802028 MDA-MB-468 Homo sapiens
## PMC42ET_Ctrl_Rep1 SRX1795057,SRX1795087,SRX1802029   PMC42-ET Homo sapiens
## PMC42ET_Ctrl_Rep2 SRX1795039,SRX1795067,SRX1802011   PMC42-ET Homo sapiens
## PMC42ET_Ctrl_Rep3 SRX1795040,SRX1795068,SRX1802012   PMC42-ET Homo sapiens
## PMC42ET_EGF_Rep1  SRX1795041,SRX1795069,SRX1802013   PMC42-ET Homo sapiens
## PMC42ET_EGF_Rep2  SRX1795042,SRX1795070,SRX1802014   PMC42-ET Homo sapiens
## PMC42ET_EGF_Rep3  SRX1795043,SRX1795071,SRX1802015   PMC42-ET Homo sapiens
## PMC42LA_Ctrl_Rep1 SRX1795044,SRX1795072,SRX1802016   PMC42-LA Homo sapiens
## PMC42LA_Ctrl_Rep2 SRX1795045,SRX1795073,SRX1802017   PMC42-LA Homo sapiens
## PMC42LA_Ctrl_Rep3 SRX1795046,SRX1795074,SRX1802018   PMC42-LA Homo sapiens
## PMC42LA_EGF_Rep1  SRX1795047,SRX1795075,SRX1802019   PMC42-LA Homo sapiens
## PMC42LA_EGF_Rep2  SRX1795048,SRX1795076,SRX1802020   PMC42-LA Homo sapiens
## PMC42LA_EGF_Rep3  SRX1795050,SRX1795079,SRX1802022   PMC42-LA Homo sapiens
##                   SRA.Study
## MDA468_Ctrl_Rep1  SRP075592
## MDA468_Ctrl_Rep2  SRP075592
## MDA468_Ctrl_Rep3  SRP075592
## MDA468_EGF_Rep1   SRP075592
## MDA468_EGF_Rep2   SRP075592
## MDA468_EGF_Rep3   SRP075592
## MDA468_HPX_Rep1   SRP075592
## MDA468_HPX_Rep2   SRP075592
## MDA468_HPX_Rep3   SRP075592
## PMC42ET_Ctrl_Rep1 SRP075592
## PMC42ET_Ctrl_Rep2 SRP075592
## PMC42ET_Ctrl_Rep3 SRP075592
## PMC42ET_EGF_Rep1  SRP075592
## PMC42ET_EGF_Rep2  SRP075592
## PMC42ET_EGF_Rep3  SRP075592
## PMC42LA_Ctrl_Rep1 SRP075592
## PMC42LA_Ctrl_Rep2 SRP075592
## PMC42LA_Ctrl_Rep3 SRP075592
## PMC42LA_EGF_Rep1  SRP075592
## PMC42LA_EGF_Rep2  SRP075592
## PMC42LA_EGF_Rep3  SRP075592</code></pre>
<p>Gene identifiers in this data are Ensembl gene identifiers, however, gene symbols or Entrez IDs will be requried for the downstream analysis. We will convert Ensembl IDs to gene symbols as these are easier to interpret visually. The row annotations of <code>emt_se</code> contain mappings to other identifiers such as gene symbols. We can convert Ensembl IDs to gene symbols by using the mappings stored in the object. Due to the issue of multi-mapping between identifiers, we first have to deal with duplicate mappings. We can identify duplicates as below.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#identify duplicated mappings</span>
<span class="va">all_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">emt_se</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span>
<span class="va">dups</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="va">dups</span></code></pre></div>
<pre><code>##  [1] "Y_RNA"       "RGS5"        "TBCE"        "LINC00486"   "Metazoa_SRP"
##  [6] "LINC01238"   "CYB561D2"    "POLR2J4"     "POLR2J3"     "5S_rRNA"    
## [11] "TMSB15B"     "ALG1L9P"     "DNAJC9-AS1"  "BMS1P4"      "SNORA70"    
## [16] "HERC2P7"     "U2"          "U3"          "ELFN2"       "5_8S_rRNA"  
## [21] "U1"</code></pre>
<p>Most duplicates are arising from miscellaneous RNA therefore duplicated mappings can either be removed or resolved manually using other annotations such as their biotypes (for example, retaining protein coding genes). For simplicity, we discard multimapped genes in this analysis. Since the number of genes being discarded is relatively small and the genes being discarded are unrelated to the process we wish to study (EMT), the effect of their removal will be minimal and can be safely ignored.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#remove duplicated genes</span>
<span class="va">emt_se</span> <span class="op">=</span> <span class="va">emt_se</span><span class="op">[</span><span class="op">!</span><span class="va">all_genes</span> <span class="op">%in%</span> <span class="va">dups</span>, <span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">emt_se</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">emt_se</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span>
<span class="va">emt_se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 29752 21 
## metadata(0):
## assays(2): counts logRPKM
## rownames(29752): DDX11L1 WASH7P ... AC004556.3 AC240274.1
## rowData names(7): Chr Start ... gene_name gene_biotype
## colnames(21): MDA468_Ctrl_Rep1 MDA468_Ctrl_Rep2 ... PMC42LA_EGF_Rep2
##   PMC42LA_EGF_Rep3
## colData names(14): group lib.size ... Organism SRA.Study</code></pre>
</div>
<div id="load-gene-sets" class="section level2">
<h2 class="hasAnchor">
<a href="#load-gene-sets" class="anchor"></a>Load gene-sets</h2>
<p>Gene-sets representing the molecular phenotypes of interest need to be prepared to query individual samples against. We can either use signatures from the molecular signatures database made available through the <code>msigdb</code> R/Bioconductor package or we can use custom signatures from publications. <span class="citation">Tan et al. (2014)</span> developed gene-sets that represent an epithelial and a mesenchymal phenotype. We will use these to explore the epithelial-mesenchymal landscape of samples. Data from the publication is organised in the <em>Thiery_EMTsignatures.txt</em> file that is stored in the package associated with this workflow.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#read Thiery et al. signatures</span>
<span class="va">thiery_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">'extdata/Thiery_EMTsignatures.txt'</span>, package <span class="op">=</span> <span class="st">'GenesetAnalysisWorkflow'</span><span class="op">)</span>
<span class="va">thiery_data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="va">thiery_path</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">thiery_data</span><span class="op">)</span></code></pre></div>
<pre><code>##   officialSymbol     genes epiMes_tumor epiMes_cellLine Ensembl.Gene.ID
## 1          ABCC3     ABCC3          epi             epi ENSG00000108846
## 2         ABHD11    ABHD11          epi            &lt;NA&gt; ENSG00000106077
## 3          ADAP1     ADAP1         &lt;NA&gt;             epi ENSG00000105963
## 4          ADIRF C10orf116         &lt;NA&gt;             epi ENSG00000148671
## 5           AGR2      AGR2          epi             epi ENSG00000106541
## 6           AIM1      AIM1         &lt;NA&gt;             epi ENSG00000112297
##   HGNC.symbol EntrezGene.ID
## 1       ABCC3          8714
## 2      ABHD11         83451
## 3       ADAP1         11033
## 4       ADIRF         10974
## 5        AGR2         10551
## 6        AIM1           202</code></pre>
<p>Since gene symbols are used in the gene expression data, we will use create gene-sets using gene symbols. We will use signatures specific to cell lines from <span class="citation">Tan et al. (2014)</span>. Gene-sets are stored in the dedicated <code><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-methods.html">GSEABase::GeneSet</a></code> object that allows storage of metadata associated with gene-sets. Singscore can work with gene-sets defined as a character vector or in GeneSet objects. Since the latter is more structured, we prefer to use them for our analysis. GeneSet objects are used in the following vissE analysis too, further motivating their use. Unique gene identifiers are required for GeneSet objects. It is good practice to specify a unique, well-defined setName for each gene-set. Additionally, to enable usage with the <code>vissE</code> package, we will specify the identifier type (i.e. <code><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html">SymbolIdentifier()</a></code>).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span>

<span class="co">#retrieve epithelial genes</span>
<span class="va">epi_genes</span> <span class="op">=</span> <span class="va">thiery_data</span><span class="op">$</span><span class="va">officialSymbol</span><span class="op">[</span><span class="va">thiery_data</span><span class="op">$</span><span class="va">epiMes_cellLine</span> <span class="op">%in%</span> <span class="st">'epi'</span><span class="op">]</span>
<span class="co">#select unique genes</span>
<span class="va">epi_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">epi_genes</span><span class="op">)</span>
<span class="co">#create GeneSet object</span>
<span class="va">epi_sig</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-methods.html">GeneSet</a></span><span class="op">(</span><span class="va">epi_genes</span>, setName <span class="op">=</span> <span class="st">'THIERY_EPITHELIAL_CELLLINE'</span>, geneIdType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">epi_sig</span></code></pre></div>
<pre><code>## setName: THIERY_EPITHELIAL_CELLLINE 
## geneIds: ABCC3, ADAP1, ..., ZNF165 (total: 170)
## geneIdType: Symbol
## collectionType: Null 
## details: use 'details(object)'</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#retrieve mesenchymal genes</span>
<span class="va">mes_genes</span> <span class="op">=</span> <span class="va">thiery_data</span><span class="op">$</span><span class="va">genes</span><span class="op">[</span><span class="va">thiery_data</span><span class="op">$</span><span class="va">epiMes_cellLine</span> <span class="op">%in%</span> <span class="st">'mes'</span><span class="op">]</span>
<span class="co">#select unique genes</span>
<span class="va">mes_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">mes_genes</span><span class="op">)</span>
<span class="co">#create GeneSet object</span>
<span class="va">mes_sig</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-methods.html">GeneSet</a></span><span class="op">(</span><span class="va">mes_genes</span>, setName <span class="op">=</span> <span class="st">'THIERY_MESENCHYMAL_CELLLINE'</span>, geneIdType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">mes_sig</span></code></pre></div>
<pre><code>## setName: THIERY_MESENCHYMAL_CELLLINE 
## geneIds: AKAP12, ANK2, ..., ZEB1 (total: 48)
## geneIdType: Symbol
## collectionType: Null 
## details: use 'details(object)'</code></pre>
<p>If the gene-sets of interest are available in the molecular signatures database (MSigDB), the <code>msigdb</code> package can be used. This package uses the <code>ExperimentHub</code> to make MSigDB gene-sets available as GeneSet objects in a <code><a href="https://rdrr.io/pkg/GSEABase/man/GeneSetCollection-methods.html">GSEABase::GeneSetCollection</a></code> object. The GeneSetCollection object can be used to store multiple GeneSet objects. The <code>msigdb</code> package provides signatures for human and mouse (the version provided by the Smyth lab at WEHI) using either gene symbols or Entrez IDs. The package provides versions of MSigDB including and greater than 7.2.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/msigdb">msigdb</a></span><span class="op">)</span>

<span class="co">#load MSigDB gene-sets</span>
<span class="va">msigdb.hs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/getMsigdb.html">getMsigdb</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">'hs'</span>, id <span class="op">=</span> <span class="st">'SYM'</span>, version <span class="op">=</span> <span class="st">'7.2'</span><span class="op">)</span>
<span class="co">#add KEGG gene-sets</span>
<span class="va">msigdb.hs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/appendKEGG.html">appendKEGG</a></span><span class="op">(</span><span class="va">msigdb.hs</span><span class="op">)</span>
<span class="va">msigdb.hs</span></code></pre></div>
<pre><code>## GeneSetCollection
##   names: chr11q, chr6q, ..., KEGG_VIRAL_MYOCARDITIS (31508 total)
##   unique identifiers: AP001767.2, SLC22A9, ..., AC023491.2 (40044 total)
##   types in collection:
##     geneIdType: SymbolIdentifier (1 total)
##     collectionType: BroadCollection (1 total)</code></pre>
</div>
<div id="singscore" class="section level2">
<h2 class="hasAnchor">
<a href="#singscore" class="anchor"></a>Singscore</h2>
<p>With the data and gene-sets ready, we can now score samples against gene-sets to explore molecular phenotypes using singscore <span class="citation">(Foroutan et al. 2018)</span>. Singscore is a rank-based single sample scoring approach implemented in the <code>singscore</code> R/Bioconductor package. The core computation of scores can be summarised in two steps: computing within sample gene ranks and computing gene-set scores using ranks. Data used for singscore should be length bias corrected (e.g. TPM, RPKM, RSEM) and should have filtered out genes with low expression across most samples. The data used here has been processed as such therefore no further pre-processing is required. Further information on length-bias correction and filtering with respect to singscore can be found in <span class="citation">Bhuva et al. (2019)</span>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/singscore">singscore</a></span><span class="op">)</span>

<span class="co">#rank genes based on expression (logRPKM)</span>
<span class="va">eranks</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/rankGenes.html">rankGenes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">emt_se</span>, <span class="st">'logRPKM'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#compute epithelial scores</span>
<span class="va">epi_score</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">eranks</span>, <span class="va">epi_sig</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">epi_score</span><span class="op">)</span></code></pre></div>
<pre><code>##                  TotalScore TotalDispersion
## MDA468_Ctrl_Rep1  0.3230845        3992.642
## MDA468_Ctrl_Rep2  0.3222157        4030.448
## MDA468_Ctrl_Rep3  0.3214232        4120.145
## MDA468_EGF_Rep1   0.3230362        4076.409
## MDA468_EGF_Rep2   0.3218094        4169.812
## MDA468_EGF_Rep3   0.3274617        3951.870</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#compute mesenchymal scores</span>
<span class="va">mes_score</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">eranks</span>, <span class="va">mes_sig</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mes_score</span><span class="op">)</span></code></pre></div>
<pre><code>##                    TotalScore TotalDispersion
## MDA468_Ctrl_Rep1 -0.027817504       10014.963
## MDA468_Ctrl_Rep2 -0.032615987       10333.722
## MDA468_Ctrl_Rep3 -0.030163516       10885.249
## MDA468_EGF_Rep1   0.014965710       10068.337
## MDA468_EGF_Rep2   0.008487366       10682.133
## MDA468_EGF_Rep3   0.018875730        8914.874</code></pre>
<p>Unlike other single-sample methods, the results from singscore contain two values, the score and the dispersion. We have found both to be useful in interpreting scores from gene-sets. The score is an indication of the median expression of all genes in the gene-set relative to all measured genes. The dispersion is a measure of the coordination of gene expression for genes in the signature. Lower dispersion indicates that genes in the gene-set are expressed at a similar level. The <code><a href="https://davislaboratory.github.io/singscore/reference/plotDispersion.html">plotDispersion()</a></code> function can be used to plot scores against their dispersions. Samples can be annotated using discrete or continous annotations. All plots in the <code>singscore</code> package can be turned into interactive plots by setting <code>isInteractive = TRUE</code>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/plotDispersion.html">plotDispersion</a></span><span class="op">(</span>
  <span class="va">epi_score</span>,
  annot <span class="op">=</span> <span class="va">emt_se</span><span class="op">$</span><span class="va">Subline</span>,
  annot_name <span class="op">=</span> <span class="st">'Subline'</span>,
  size <span class="op">=</span> <span class="fl">5</span>,
  alpha <span class="op">=</span> <span class="fl">0.5</span>,
  isInteractive <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>The plot above shows that the MDA-MB-468 cell line is more epithelial than the other cell lines as indicated by larger scores. Lower dispersion values for these samples indicate that epithelial genes are expressed at similar levels relative to other genes. The mesenchymal signature shows the inverse pattern with PMC42-ET and PMC42-LA cell lines having higher scores and lower dispersion values compared to MDA-MB-468.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/plotDispersion.html">plotDispersion</a></span><span class="op">(</span>
  <span class="va">mes_score</span>,
  annot <span class="op">=</span> <span class="va">emt_se</span><span class="op">$</span><span class="va">Subline</span>,
  annot_name <span class="op">=</span> <span class="st">'Subline'</span>,
  size <span class="op">=</span> <span class="fl">5</span>,
  alpha <span class="op">=</span> <span class="fl">0.5</span>,
  isInteractive <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>It is often useful to investigate the association between gene-sets, especially when phenotypes are expected to be associated biologically. <span class="citation">Cursons et al. (2018)</span> and <span class="citation">Foroutan et al. (2018)</span> have used the epithelial-mesenchymal landscapes to better understand the behaviour of biological models across the epithelial mesenchymal transition. Additionally, it is often the case in biology that biological models are compared with patient data to enable translation of discoveries. Singscore enables this by plotting cell line data on a background dataset. In the plots below, we plot breast cancer patient data from The Cancer Genome Atlas (TCGA) in the background. We then project our cell line data onto the patient data to explore and explain various aspects of the epithelial mesenchymal landscape. Projecting cell line data onto the patient data enables transfer of discoveries made on cell lines to patients. We use pre-computed epithelial and mesenchymal scores of TCGA breast cancer patient data from the <code>singscore</code> package.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#load pre-computed TCGA breast cancer EMT scores</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"scoredf_tcga_epi"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"scoredf_tcga_mes"</span><span class="op">)</span>

<span class="co">#plot an EMT landscape</span>
<span class="va">p_tcga</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/plotScoreLandscape.html">plotScoreLandscape</a></span><span class="op">(</span>
  scoredf1 <span class="op">=</span> <span class="va">scoredf_tcga_epi</span>,
  scoredf2 <span class="op">=</span> <span class="va">scoredf_tcga_mes</span>,
  scorenames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Epithelial score'</span>, <span class="st">'Mesenchymal score'</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p>The plot below shows that PMC42-ET and PMC42-LA have similar mesenchymal scores, but very distinct epithelial scores. MDA-MB-468 on the other hand is a very epithelial cell line and less mesenchymal. We would suspect that PMC42-LA has both epithelial and mesenchymal properties. Like other plots in singscore, the plots below can be made interactive setting <code>isInteractive = TRUE</code>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/projectScoreLandscape.html">projectScoreLandscape</a></span><span class="op">(</span>
  <span class="va">p_tcga</span>,
  scoredf1 <span class="op">=</span> <span class="va">epi_score</span>,
  scoredf2 <span class="op">=</span> <span class="va">mes_score</span>,
  annot <span class="op">=</span> <span class="va">emt_se</span><span class="op">$</span><span class="va">Subline</span>,
  annot_name <span class="op">=</span> <span class="st">'Subline'</span>,
  isInteractive <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>Annotating samples using the treatment shows that hypoxic treatment of MDA-MB-468 results in increased mesenchymal properties of the cell line.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/projectScoreLandscape.html">projectScoreLandscape</a></span><span class="op">(</span>
  <span class="va">p_tcga</span>,
  scoredf1 <span class="op">=</span> <span class="va">epi_score</span>,
  scoredf2 <span class="op">=</span> <span class="va">mes_score</span>,
  annot <span class="op">=</span> <span class="va">emt_se</span><span class="op">$</span><span class="va">Treatment</span>,
  annot_name <span class="op">=</span> <span class="st">'Treatment'</span>,
  isInteractive <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p>The Hallmark collection of the MSigDB represent key molecular processes associated with cancer progression. The HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION represents EMT in cancer therefore we could compute scores for samples using it. We can annotate samples using the hallmark EMT score to assess their relation with the EMT landscape. The plot below shows that the combined epithelial-mesenchymal axis is represented by the hallmark EMT signature.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#extract the hallmark EMT signature</span>
<span class="va">hemt_sig</span> <span class="op">=</span> <span class="va">msigdb.hs</span><span class="op">[[</span><span class="st">'HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION'</span><span class="op">]</span><span class="op">]</span>
<span class="co">#score samples using the hallmark EMT signature</span>
<span class="va">hemt_score</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">eranks</span>, <span class="va">hemt_sig</span><span class="op">)</span>

<span class="co">#plot Hallmark EMT signature</span>
<span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/projectScoreLandscape.html">projectScoreLandscape</a></span><span class="op">(</span>
  <span class="va">p_tcga</span>,
  scoredf1 <span class="op">=</span> <span class="va">epi_score</span>,
  scoredf2 <span class="op">=</span> <span class="va">mes_score</span>,
  annot <span class="op">=</span> <span class="va">hemt_score</span><span class="op">$</span><span class="va">TotalScore</span>,
  annot_name <span class="op">=</span> <span class="st">'Hallmark EMT score'</span>,
  isInteractive <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<p>The above plots show that the difference between PMC42-ET and PMC42-LA is mainly along the epithelial axis. We can investigate the contribution of each gene across individual samples using the <code><a href="https://davislaboratory.github.io/singscore/reference/plotRankDensity.html">plotRankDensity()</a></code> function. The distribution of unit normalised ranks shows that most genes in the epithelial gene-set are expressed at higher levels that all other genes in PMC42-LA. In contrast, epithelial genes are more dispersed in PMC42-ET. Interactive versions of these plots can be used to explore individual genes.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/plotRankDensity.html">plotRankDensity</a></span><span class="op">(</span><span class="va">eranks</span><span class="op">[</span>, <span class="st">'PMC42LA_Ctrl_Rep1'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, <span class="va">epi_sig</span>, isInteractive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/plotRankDensity.html">plotRankDensity</a></span><span class="op">(</span><span class="va">eranks</span><span class="op">[</span>, <span class="st">'PMC42ET_Ctrl_Rep1'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, <span class="va">epi_sig</span>, isInteractive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-17-2.png" width="700"></p>
</div>
<div id="stingscore" class="section level2">
<h2 class="hasAnchor">
<a href="#stingscore" class="anchor"></a>Stingscore</h2>
<p>At times whole transcriptome measurements are not feasible either due to costs or due to availability of high quality genomic material (as is the case with FFPE samples). In such cases, we may wish to work with targeted transcriptomic panels. Most gene-set scoring methods, including singscore, require transcriptome-wide measurements therefore making them unusable in a targeted sequencing context. In such cases, the stingscore method <span class="citation">(Bhuva, Cursons, and Davis 2020)</span> implemented in the <code>singscore</code> package can be used. Stingscore makes use of stably expressed genes to estimate gene-set scores using a reduced set of measurements.</p>
<p>To demonstrate usage of stingscore in a targeted seqencing context, we will simulate a targeted sequencing panel using epithelial genes, mesenchymal genes and a few stably expressed genes. We will use the top 5 stably expressed genes identified in <span class="citation">Bhuva, Cursons, and Davis (2020)</span> to score samples.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#simulate data by sub-sampling</span>
<span class="va">sample_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">union</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html">geneIds</a></span><span class="op">(</span><span class="va">epi_sig</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html">geneIds</a></span><span class="op">(</span><span class="va">mes_sig</span><span class="op">)</span><span class="op">)</span>
<span class="va">sample_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">union</a></span><span class="op">(</span><span class="va">sample_genes</span>, <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/getStableGenes.html">getStableGenes</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span>
<span class="co">#subset genes with measurements in the data</span>
<span class="va">sample_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="va">sample_genes</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">emt_se</span><span class="op">)</span><span class="op">)</span>
<span class="co">#subset data</span>
<span class="va">targeted_se</span> <span class="op">=</span> <span class="va">emt_se</span><span class="op">[</span><span class="va">sample_genes</span>, <span class="op">]</span>
<span class="va">targeted_se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 214 21 
## metadata(0):
## assays(2): counts logRPKM
## rownames(214): ABCC3 ADAP1 ... TARDBP HNRNPK
## rowData names(7): Chr Start ... gene_name gene_biotype
## colnames(21): MDA468_Ctrl_Rep1 MDA468_Ctrl_Rep2 ... PMC42LA_EGF_Rep2
##   PMC42LA_EGF_Rep3
## colData names(14): group lib.size ... Organism SRA.Study</code></pre>
<p>The data samples genes of interest from the full data matrix resulting in the measurement of 214 genes. In reality, a panel of this size could be measure using technologies such as NanoString’s nCounter panel. The top 5 stably expressed genes used here could be used to score samples and to normalise expression measurements for additional analyses. The only difference between singscore and stingscore are in the way ranks are computed. In stingscore, ranks are estimated using stably expressed genes. Following rank estimation, scores can be computed as with singscore using the <code><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore()</a></code> function.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">st_genes</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/getStableGenes.html">getStableGenes</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span>
<span class="va">st_genes</span></code></pre></div>
<pre><code>## [1] "RBM45"  "BRAP"   "CIAO1"  "TARDBP" "HNRNPK"</code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#rank genes using stably expressed genes</span>
<span class="va">st_eranks</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/rankGenes.html">rankGenes</a></span><span class="op">(</span><span class="va">targeted_se</span>, stableGenes <span class="op">=</span> <span class="va">st_genes</span><span class="op">)</span>

<span class="co">#score samples using the simulated targeted sequencing data</span>
<span class="va">epi_score_st</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">st_eranks</span>, <span class="va">epi_sig</span><span class="op">)</span>
<span class="va">mes_score_st</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">st_eranks</span>, <span class="va">mes_sig</span><span class="op">)</span>

<span class="co">#plot scores computed using singscore vs stingscore</span>
<span class="va">colmap</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'MDA-MB-468'</span> <span class="op">=</span> <span class="st">'#1B9E77'</span>, <span class="st">'PMC42-ET'</span> <span class="op">=</span> <span class="st">'#D95F02'</span>, <span class="st">'PMC42-LA'</span> <span class="op">=</span> <span class="st">'#7570B3'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span>
  <span class="va">epi_score</span><span class="op">$</span><span class="va">TotalScore</span>,
  <span class="va">epi_score_st</span><span class="op">$</span><span class="va">TotalScore</span>,
  col <span class="op">=</span> <span class="va">colmap</span><span class="op">[</span><span class="va">targeted_se</span><span class="op">$</span><span class="va">Subline</span><span class="op">]</span>,
  main <span class="op">=</span> <span class="st">'Epithelial score'</span>,
  xlab <span class="op">=</span> <span class="st">'Singscore'</span>,
  ylab <span class="op">=</span> <span class="st">'Stingscore (targeted)'</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>coef <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, col <span class="op">=</span> <span class="fl">2</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span>
  <span class="va">mes_score</span><span class="op">$</span><span class="va">TotalScore</span>,
  <span class="va">mes_score_st</span><span class="op">$</span><span class="va">TotalScore</span>,
  col <span class="op">=</span> <span class="va">colmap</span><span class="op">[</span><span class="va">targeted_se</span><span class="op">$</span><span class="va">Subline</span><span class="op">]</span>,
  main <span class="op">=</span> <span class="st">'Mesenchymal score'</span>,
  xlab <span class="op">=</span> <span class="st">'Singscore'</span>,
  ylab <span class="op">=</span> <span class="st">'Stingscore (targeted)'</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>coef <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, col <span class="op">=</span> <span class="fl">2</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<p>The plots above show that both epithelial and mesenchymal scores computed using singscore and stingscore are highly correlated. Though the scores are correlated, their values are not an exact match as there is an offset. As the relative order of samples remains the same, biological inferences are unaffected by the offset in scores computed using stingscore.</p>
</div>
<div id="multiscore" class="section level2">
<h2 class="hasAnchor">
<a href="#multiscore" class="anchor"></a>Multiscore</h2>
<p>When the hypothesis are not pre-determined, we may wish to perform an exploratory analysis using multiple gene-sets. In such a scenario, we would wish to score samples against 100s-1000s of gene-sets which we would then summarise using further downstream analysis. The <code><a href="https://davislaboratory.github.io/singscore/reference/multiScore.html">multiScore()</a></code> function can be used to score samples against multiple signatures simultaneously. This implementation is much faster than <code><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore()</a></code> and is therefore the recommended function for scoring against multiple gene-sets.</p>
<p>We will score samples against gene-sets from the hallmark collection, gene ontologies, KEGG pathways, Reactome pathways and Biocarta pathways. The <code><a href="https://rdrr.io/pkg/msigdb/man/subsetCollection.html">subsetCollection()</a></code> function in the <code>msigdb</code> package can be used to subset a GeneSetCollection based on MSigDB categories and sub-categories. The function returns GeneSetCollection objects which can be concatenated as normal lists. The resulting list needs to be converted back into a GeneSetCollection object before passing it to <code><a href="https://davislaboratory.github.io/singscore/reference/multiScore.html">multiScore()</a></code>.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#subset collections and subcollections of interest</span>
<span class="va">genesigs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
  <span class="va">epi_sig</span>,
  <span class="va">mes_sig</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/subsetCollection.html">subsetCollection</a></span><span class="op">(</span><span class="va">msigdb.hs</span>, <span class="st">'h'</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/subsetCollection.html">subsetCollection</a></span><span class="op">(</span><span class="va">msigdb.hs</span>, <span class="st">'c2'</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'CP:KEGG'</span>, <span class="st">'CP:REACTOME'</span>, <span class="st">'CP:BIOCARTA'</span><span class="op">)</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/subsetCollection.html">subsetCollection</a></span><span class="op">(</span><span class="va">msigdb.hs</span>, <span class="st">'c5'</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'GO:BP'</span>, <span class="st">'GO:MF'</span>, <span class="st">'GO:CC'</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">genesigs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSetCollection-methods.html">GeneSetCollection</a></span><span class="op">(</span><span class="va">genesigs</span><span class="op">)</span></code></pre></div>
<p>The results of <code><a href="https://davislaboratory.github.io/singscore/reference/multiScore.html">multiScore()</a></code> is a list containing a matrix of scores and a matrix of dispersion values. As some gene-sets could be entirely composed of genes not measured in the data, there may be missing values in these matrices.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">eranks</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/rankGenes.html">rankGenes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">emt_se</span>, <span class="st">'logRPKM'</span><span class="op">)</span><span class="op">)</span>
<span class="va">msigdb_scores</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/multiScore.html">multiScore</a></span><span class="op">(</span><span class="va">eranks</span>, <span class="va">genesigs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">msigdb_scores</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## $Scores
##                                  MDA468_Ctrl_Rep1 MDA468_Ctrl_Rep2
## THIERY_EPITHELIAL_CELLLINE              0.3230845       0.32221565
## THIERY_MESENCHYMAL_CELLLINE            -0.0278175      -0.03261599
## HALLMARK_TNFA_SIGNALING_VIA_NFKB        0.1893111       0.18454327
## HALLMARK_HYPOXIA                        0.1549132       0.15766930
## HALLMARK_CHOLESTEROL_HOMEOSTASIS        0.3035149       0.30298773
## 
## $Dispersions
##                                  MDA468_Ctrl_Rep1 MDA468_Ctrl_Rep2
## THIERY_EPITHELIAL_CELLLINE               3992.642         4030.448
## THIERY_MESENCHYMAL_CELLLINE             10014.963        10333.722
## HALLMARK_TNFA_SIGNALING_VIA_NFKB         7278.825         7191.351
## HALLMARK_HYPOXIA                         7614.634         7633.907
## HALLMARK_CHOLESTEROL_HOMEOSTASIS         4222.445         4188.345</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">msigdb_scores</span>, <span class="va">dim</span><span class="op">)</span></code></pre></div>
<pre><code>## $Scores
## [1] 12352    21
## 
## $Dispersions
## [1] 12352    21</code></pre>
<p>Plotting scores against dispersion values across all signatures demonstrates the usual trend between these two values. The trend is expected since strong positive or negative scores can only be obtained if genes in gene-sets are coordinately expressed. The more interesting gene-sets are those that have low dispersion with scores close to 0; these gene-sets contain coordinately expressed genes with average expression relative to all measured genes.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/smoothScatter.html">smoothScatter</a></span><span class="op">(</span><span class="va">msigdb_scores</span><span class="op">$</span><span class="va">Scores</span>, <span class="va">msigdb_scores</span><span class="op">$</span><span class="va">Dispersions</span>, xlab <span class="op">=</span> <span class="st">'Score'</span>, ylab <span class="op">=</span> <span class="st">'Dispersion'</span>, main <span class="op">=</span> <span class="st">'Score vs Dispersion'</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-22-1.png" width="700"></p>
<p>We can assess gene-sets for individual samples to determine their phenotype composition. As transcriptomic measurements are more biologically meaningful when assessed relatively, we compute the difference in scores between one PMC42-ET control replicate and another PMC42-LA control replicate. The top positive and negative scores are listed below.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#select scores for a single Hypoxic sample</span>
<span class="va">et_la_scores</span> <span class="op">=</span> <span class="va">msigdb_scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">[</span>, <span class="st">'PMC42ET_Ctrl_Rep1'</span><span class="op">]</span> <span class="op">-</span> <span class="va">msigdb_scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">[</span>, <span class="st">'PMC42LA_Ctrl_Rep1'</span><span class="op">]</span>
<span class="va">et_la_scores</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">et_la_scores</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">et_la_scores</span><span class="op">)</span></code></pre></div>
<pre><code>##                                                 GO_REGULATION_OF_EPINEPHRINE_SECRETION 
##                                                                             -0.3560168 
##                                           GO_POSITIVE_REGULATION_OF_MESODERM_FORMATION 
##                                                                             -0.3180387 
##                      GO_REGULATION_OF_VERY_LOW_DENSITY_LIPOPROTEIN_PARTICLE_REMODELING 
##                                                                             -0.3061849 
## GO_GAP_JUNCTION_CHANNEL_ACTIVITY_INVOLVED_IN_CELL_COMMUNICATION_BY_ELECTRICAL_COUPLING 
##                                                                             -0.3041113 
##                                           GO_INTERMEDIATE_DENSITY_LIPOPROTEIN_PARTICLE 
##                                                                             -0.2954161 
##                                                GO_NORADRENERGIC_NEURON_DIFFERENTIATION 
##                                                                             -0.2788329</code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="va">et_la_scores</span><span class="op">)</span></code></pre></div>
<pre><code>##                            BIOCARTA_CYTOKINE_PATHWAY 
##                                            0.2212715 
## GO_POSITIVE_REGULATION_OF_T_CELL_TOLERANCE_INDUCTION 
##                                            0.2223007 
##  GO_CORTICOTROPIN_RELEASING_HORMONE_RECEPTOR_BINDING 
##                                            0.2240487 
##          GO_REGULATION_OF_LIPID_TRANSPORTER_ACTIVITY 
##                                            0.2332605 
##                            GO_11_CIS_RETINAL_BINDING 
##                                            0.2378071 
##                   GO_ESTROGEN_2_HYDROXYLASE_ACTIVITY 
##                                            0.2410252</code></pre>
</div>
</div>
<div id="identifying-and-visualising-higher-order-phenotypes" class="section level1">
<h1 class="hasAnchor">
<a href="#identifying-and-visualising-higher-order-phenotypes" class="anchor"></a>Identifying and visualising higher-order phenotypes</h1>
<div id="perform-de-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#perform-de-analysis" class="anchor"></a>Perform DE analysis</h2>
<p>We perform DE analysis using the quasi-likelihood edgeR pipeline <span class="citation">(Chen, Lun, and Smyth 2016)</span>.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma">limma</a></span><span class="op">)</span>

<span class="co">#subset dataset to extract only control samples of PMC42-ET and PMC42-LA</span>
<span class="va">emt_se_sub</span> <span class="op">=</span> <span class="va">emt_se</span><span class="op">[</span>, <span class="va">emt_se</span><span class="op">$</span><span class="va">Subline</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'PMC42-ET'</span>, <span class="st">'PMC42-LA'</span><span class="op">)</span> <span class="op">&amp;</span>
                      <span class="va">emt_se</span><span class="op">$</span><span class="va">Treatment</span> <span class="op">==</span> <span class="st">'control'</span><span class="op">]</span>
<span class="co">#create DGElist object from SE object</span>
<span class="va">emt_dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/emtdata/man/asDGEList.html">asDGEList</a></span><span class="op">(</span><span class="va">emt_se_sub</span><span class="op">)</span>
<span class="co">#create model design matrix</span>
<span class="va">design</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Subline</span>, data <span class="op">=</span> <span class="va">emt_dge</span><span class="op">$</span><span class="va">samples</span><span class="op">)</span>
<span class="co">#remove lowly expressed genes</span>
<span class="va">keep</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html">filterByExpr</a></span><span class="op">(</span><span class="va">emt_dge</span>, design <span class="op">=</span> <span class="va">design</span><span class="op">)</span>
<span class="va">emt_dge</span> <span class="op">=</span> <span class="va">emt_dge</span><span class="op">[</span><span class="va">keep</span>, <span class="op">]</span>
<span class="co">#calculate normalisation factor using on TMM</span>
<span class="va">emt_dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span><span class="op">(</span><span class="va">emt_dge</span>, method <span class="op">=</span> <span class="st">"TMM"</span><span class="op">)</span>

<span class="co">#estimate dispersions</span>
<span class="va">emt_dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateDisp.html">estimateDisp</a></span><span class="op">(</span><span class="va">emt_dge</span>, design <span class="op">=</span> <span class="va">design</span>, robust <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#fit usimg edgeR's Generalized Linear Models with Quasi-likelihood Tests (GLM) function</span>
<span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html">glmQLFit</a></span><span class="op">(</span><span class="va">emt_dge</span>, <span class="va">design</span>, robust <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#get DE genes list</span>
<span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html">glmQLFTest</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="st">'SublinePMC42-LA'</span><span class="op">)</span>
<span class="va">is.de</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/decidetestsDGE.html">decideTestsDGE</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMD.html">plotMD</a></span><span class="op">(</span><span class="va">fit</span>, status <span class="op">=</span> <span class="va">is.de</span>, main <span class="op">=</span> <span class="st">'PMC42-LA vs. PMC42-ET'</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get DE list based on p-value 0.05 cutoff</span>
<span class="va">DE_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html">topTags</a></span><span class="op">(</span><span class="va">res</span>, n <span class="op">=</span> <span class="cn">Inf</span>, p.value <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">DE_list</span><span class="op">)</span></code></pre></div>
<pre><code>##          gene_biotype     logFC   logCPM         F       PValue          FDR
## FAT2   protein_coding -7.588511 6.614230 1334.7019 5.576353e-13 1.346410e-08
## TGFBR2 protein_coding -9.993858 3.596306 1040.0713 2.218916e-12 1.759046e-08
## SPINT1 protein_coding  6.890147 5.922593  985.9245 2.982160e-12 1.759046e-08
## PLS3   protein_coding -6.059286 5.070152  958.2729 3.489918e-12 1.759046e-08
## PLTP   protein_coding -6.202561 5.935153  950.8729 3.642671e-12 1.759046e-08
## KLK11  protein_coding  8.779300 3.708962  877.8718 5.662833e-12 2.278818e-08</code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get all genes statistics</span>
<span class="va">allg_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html">topTags</a></span><span class="op">(</span><span class="va">res</span>, n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
</div>
<div id="gene-set-testing-using-limmafry" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-set-testing-using-limmafry" class="anchor"></a>Gene-set testing using <code>limma::fry</code>
</h2>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Use geneset collection (genesigs) generated earlier to create fry indices</span>
<span class="va">fry_indices</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ids2indices.html">ids2indices</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html">geneIds</a></span><span class="op">(</span><span class="va">genesigs</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">emt_dge</span><span class="op">)</span><span class="op">)</span>
<span class="co">#remove very small and very large genesets</span>
<span class="va">fry_indices</span> <span class="op">=</span> <span class="va">fry_indices</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">fry_indices</span>, <span class="va">length</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">10</span>
                          <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">fry_indices</span>, <span class="va">length</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">500</span><span class="op">]</span>

<span class="co">#Perform a geneset enrichment analysis using limma::fry</span>
<span class="va">fry_res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/roast.html">fry</a></span><span class="op">(</span><span class="va">emt_dge</span>,
              index <span class="op">=</span> <span class="va">fry_indices</span>,
              design <span class="op">=</span> <span class="va">design</span>,
              contrast <span class="op">=</span> <span class="st">'SublinePMC42-LA'</span><span class="op">)</span>

<span class="co">#select significant results (FDR &lt; 0.05)</span>
<span class="va">fry_res</span> <span class="op">=</span> <span class="va">fry_res</span><span class="op">[</span><span class="va">fry_res</span><span class="op">$</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">fry_res</span><span class="op">)</span></code></pre></div>
<pre><code>##                                                         NGenes Direction
## GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS                     13        Up
## GO_LABYRINTHINE_LAYER_MORPHOGENESIS                         18        Up
## GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION                16      Down
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION     15        Up
## GO_DNA_POLYMERASE_COMPLEX                                   21      Down
## REACTOME_RA_BIOSYNTHESIS_PATHWAY                            16        Up
##                                                               PValue
## GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS                 7.343476e-07
## GO_LABYRINTHINE_LAYER_MORPHOGENESIS                     2.604803e-06
## GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION            3.526290e-06
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION 3.787180e-06
## GO_DNA_POLYMERASE_COMPLEX                               4.208996e-06
## REACTOME_RA_BIOSYNTHESIS_PATHWAY                        4.238159e-06
##                                                                 FDR
## GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS                 0.004489003
## GO_LABYRINTHINE_LAYER_MORPHOGENESIS                     0.004489003
## GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION            0.004489003
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION 0.004489003
## GO_DNA_POLYMERASE_COMPLEX                               0.004489003
## REACTOME_RA_BIOSYNTHESIS_PATHWAY                        0.004489003
##                                                         PValue.Mixed
## GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS                 1.220636e-05
## GO_LABYRINTHINE_LAYER_MORPHOGENESIS                     3.708566e-07
## GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION            5.533273e-07
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION 3.109981e-06
## GO_DNA_POLYMERASE_COMPLEX                               6.086205e-06
## REACTOME_RA_BIOSYNTHESIS_PATHWAY                        3.192814e-06
##                                                            FDR.Mixed
## GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS                 1.980857e-05
## GO_LABYRINTHINE_LAYER_MORPHOGENESIS                     3.173336e-06
## GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION            3.608736e-06
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION 7.916704e-06
## GO_DNA_POLYMERASE_COMPLEX                               1.208747e-05
## REACTOME_RA_BIOSYNTHESIS_PATHWAY                        8.019966e-06</code></pre>
</div>
<div id="prepare-gene-set-testing-results-for-visse" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-gene-set-testing-results-for-visse" class="anchor"></a>Prepare gene-set testing results for vissE</h2>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#create a GeneSetCollection for the significant gene-sets</span>
<span class="va">siggs</span> <span class="op">=</span> <span class="va">genesigs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">fry_res</span><span class="op">)</span><span class="op">]</span>
<span class="va">siggs</span></code></pre></div>
<pre><code>## GeneSetCollection
##   names: GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS, GO_LABYRINTHINE_LAYER_MORPHOGENESIS, ..., GO_PLATELET_ALPHA_GRANULE_LUMEN (1667 total)
##   unique identifiers: CS, SIRT4, ..., CDY2A (13779 total)
##   types in collection:
##     geneIdType: SymbolIdentifier (1 total)
##     collectionType: BroadCollection, NullCollection (2 total)</code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#prepare gene-set statistics for visualisation (signed -log10(FDR))</span>
<span class="va">gsStats</span> <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">fry_res</span><span class="op">$</span><span class="va">FDR</span><span class="op">)</span>
<span class="co">#use a positive sign to indicate up-regulated gene-sets</span>
<span class="va">gsStats</span> <span class="op">=</span> <span class="va">gsStats</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Up'</span> <span class="op">=</span> <span class="fl">1</span>, <span class="st">'Down'</span> <span class="op">=</span> <span class="op">-</span><span class="fl">1</span><span class="op">)</span><span class="op">[</span><span class="va">fry_res</span><span class="op">$</span><span class="va">Direction</span><span class="op">]</span>
<span class="co">#has to be a named vector</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gsStats</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">fry_res</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gsStats</span><span class="op">)</span></code></pre></div>
<pre><code>##                 GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS 
##                                                 2.34785 
##                     GO_LABYRINTHINE_LAYER_MORPHOGENESIS 
##                                                 2.34785 
##            GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION 
##                                                -2.34785 
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION 
##                                                 2.34785 
##                               GO_DNA_POLYMERASE_COMPLEX 
##                                                -2.34785 
##                        REACTOME_RA_BIOSYNTHESIS_PATHWAY 
##                                                 2.34785</code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#prepare gene-level statistics for visualisation (logFC)</span>
<span class="va">gStats</span> <span class="op">=</span> <span class="va">allg_list</span><span class="op">$</span><span class="va">logFC</span>
<span class="co">#has to be a named vector</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gStats</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">allg_list</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gStats</span><span class="op">)</span></code></pre></div>
<pre><code>##      FAT2    TGFBR2    SPINT1      PLS3      PLTP     KLK11 
## -7.588511 -9.993858  6.890147 -6.059286 -6.202561  8.779300</code></pre>
</div>
<div id="compute-the-overlap-network" class="section level2">
<h2 class="hasAnchor">
<a href="#compute-the-overlap-network" class="anchor"></a>Compute the overlap network</h2>
<p>The default approach to computing overlaps is using the Jaccard index. Overlap is computed based on the gene overlap between gene-sets. Alternatively, the overlap coefficient can be used. The latter can be used to highlight hierarchical overlaps (such as those present in the gene ontology).</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/vissE">vissE</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org">igraph</a></span><span class="op">)</span>

<span class="co">#compute gene-set overlaps between significant gene-sets</span>
<span class="va">gs_ovlap</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/computeMsigOverlap.html">computeMsigOverlap</a></span><span class="op">(</span><span class="va">siggs</span>, thresh <span class="op">=</span> <span class="fl">0.25</span>, measure <span class="op">=</span> <span class="st">'jaccard'</span><span class="op">)</span>
<span class="co">#create a network from overlaps</span>
<span class="va">gs_ovnet</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/computeMsigNetwork.html">computeMsigNetwork</a></span><span class="op">(</span><span class="va">gs_ovlap</span>, msigGsc <span class="op">=</span> <span class="va">siggs</span><span class="op">)</span>
<span class="co">#plot the network</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">21</span><span class="op">)</span> <span class="co">#set seed for reproducible layout</span>
<span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigNetwork.html">plotMsigNetwork</a></span><span class="op">(</span><span class="va">gs_ovnet</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-27-1.png" width="768"></p>
<p>The overlap network plot above is annotated based on the MSigDB categories used. We can annotate it with a gene-level statistic. Here we use a signed -log10(FDR) where the sign represents the direction of change.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#plot the network and overlay gene-set statistics</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">21</span><span class="op">)</span> <span class="co">#set seed for reproducible layout</span>
<span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigNetwork.html">plotMsigNetwork</a></span><span class="op">(</span><span class="va">gs_ovnet</span>, genesetStat <span class="op">=</span> <span class="va">gsStats</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-28-1.png" width="768"></p>
</div>
<div id="identify-communities-within-the-network" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-communities-within-the-network" class="anchor"></a>Identify communities within the network</h2>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#identify clusters using a graph clustering algorithm</span>
<span class="va">grps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cluster_walktrap.html">cluster_walktrap</a></span><span class="op">(</span><span class="va">gs_ovnet</span><span class="op">)</span>
<span class="co">#extract clustering results</span>
<span class="va">grps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/groups.html">groups</a></span><span class="op">(</span><span class="va">grps</span><span class="op">)</span>
<span class="co">#remove small groups</span>
<span class="va">grps</span> <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">grps</span>, <span class="va">length</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">5</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">grps</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 43</code></pre>
<p>Since we could have numerous gene-set clusters, we need to come up with a prioritisation scheme to explore them. We maximise the cluster size and statistic using the product-of-ranks approach to achieve this.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#compute cluster sizes</span>
<span class="va">grp_size</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">grps</span>, <span class="va">length</span><span class="op">)</span>
<span class="co">#compute cluster statistic - median of the absolute statistic</span>
<span class="va">grp_stat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">grps</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">gsStats</span><span class="op">[</span><span class="va">x</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#combine using product of ranks</span>
<span class="va">grp_pr</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rank.html">rank</a></span><span class="op">(</span><span class="va">grp_size</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/rank.html">rank</a></span><span class="op">(</span><span class="va">grp_stat</span><span class="op">)</span>
<span class="co">#order group using the product of ranks - maximise cluster size and statistic</span>
<span class="va">grps</span> <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">grp_pr</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<p>Visualising all gene-set clusters together is difficult so we will focus on the top 12 prioritised using the product-of-ranks approach.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#plot the top 12 clusters</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">21</span><span class="op">)</span> <span class="co">#set seed for reproducible layout</span>
<span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigNetwork.html">plotMsigNetwork</a></span><span class="op">(</span><span class="va">gs_ovnet</span>, markGroups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>, genesetStat <span class="op">=</span> <span class="va">gsStats</span><span class="op">)</span>
<span class="va">p1</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-31-1.png" width="768"></p>
</div>
<div id="characterise-gene-set-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#characterise-gene-set-clusters" class="anchor"></a>Characterise gene-set clusters</h2>
<p>Gene-set clusters identified can be assessed for their biological similarities using text-mining approaches. Here, we perform a frequency analysis (adjusted for using the inverse document frequency) on the gene-set names or their short descriptions to assess recurring biological themes in clusters. These results are then presented as word clouds.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#all gene-sets have names</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">siggs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, <span class="va">setName</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS"     
## [2] "GO_LABYRINTHINE_LAYER_MORPHOGENESIS"         
## [3] "GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION"</code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#almost all gene-sets have short descriptions</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">siggs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, <span class="va">description</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "The chemical reactions and pathways involving dicarboxylic acids, any organic acid containing three carboxyl (COOH) groups or anions (COO-). [GOC:mah]"                                        
## [2] "The process in which the labyrinthine layer of the placenta is generated and organized. [GOC:dph]"                                                                                             
## [3] "The process in which interchain hydrogen bonds between two strands of DNA are broken or 'melted', generating unpaired template strands for DNA replication. [ISBN:071673706X, ISBN:0815316194]"</code></pre>
<p>These can be used to automatically annotate gene-sets</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#compute and plot the results of text-mining: using gene-set name</span>
<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigWordcloud.html">plotMsigWordcloud</a></span><span class="op">(</span>msigGsc <span class="op">=</span> <span class="va">siggs</span>, groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">'Name'</span><span class="op">)</span>
<span class="va">p2</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-33-1.png" width="864"></p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#compute and plot the results of text-mining: using gene-set short description</span>
<span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigWordcloud.html">plotMsigWordcloud</a></span><span class="op">(</span>msigGsc <span class="op">=</span> <span class="va">siggs</span>, groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">'Short'</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-33-2.png" width="864"></p>
</div>
<div id="visualise-gene-level-statistics-for-gene-set-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#visualise-gene-level-statistics-for-gene-set-clusters" class="anchor"></a>Visualise gene-level statistics for gene-set clusters</h2>
<p>Gene-level statistics for each gene-set cluster can be visualised to better understand the genes contributing to significance of gene-sets. Gene-level statistics can be passed onto the plotting function as a named vector. A jitter is applied on the x-axis (due to its discrete nature).</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#plot the gene-level statistics (logFC)</span>
<span class="va">p3</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotGeneStats.html">plotGeneStats</a></span><span class="op">(</span><span class="va">gStats</span>,
                   msigGsc <span class="op">=</span> <span class="va">siggs</span>,
                   groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>,
                   statName <span class="op">=</span> <span class="st">'logFC'</span><span class="op">)</span>
<span class="co">#since all vissE plots are ggplot2 objects, they can be modified</span>
<span class="va">p3</span> <span class="op">=</span> <span class="va">p3</span> <span class="op">+</span> <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, colour <span class="op">=</span> <span class="fl">2</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="va">p3</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-34-1.png" width="864"></p>
</div>
<div id="interpreting-biology-from-visse" class="section level2">
<h2 class="hasAnchor">
<a href="#interpreting-biology-from-visse" class="anchor"></a>Interpreting biology from vissE</h2>
<p>We can combine results from all three plots to provide a holistic view of the data.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span>

<span class="co">#combine plots into a unified visualisation</span>
<span class="va">p2</span> <span class="op">+</span> <span class="va">p1</span> <span class="op">+</span> <span class="va">p3</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-35-1.png" width="1440"></p>
<p>Cluster 90 is going up in PMC42-LA compared to PMC42-ET and is tightly clustered.<br>
Keratin associated genes (<em>KRT-x</em>) are well represented in this cluster.<br><em>GRHL2</em> (Transcription Factor Grainyhead-like 2) is suppresses TGF-beta induced EMT. Regulates epithelial morphogenesis acting as a target gene-associated transcriptional activator of apical junctional complex components.<br><em>OVOL2</em> (Ovo Like Zinc Finger 2) is a zinc-finger transcription repressor factor which induces mesenchymal-to-epithelial (MET) transition.<br><em>CDH-3</em> (P-cadherin) potentially identify an “intermediate/metastable” EMT state.</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#investigate specific clusters</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">21</span><span class="op">)</span> <span class="co">#set seed for reproducible layout</span>
<span class="co">#select group to plot</span>
<span class="va">g</span> <span class="op">=</span> <span class="st">'90'</span>
<span class="co">#compute and plot vissE panels for the group</span>
<span class="va">p1_g</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigNetwork.html">plotMsigNetwork</a></span><span class="op">(</span><span class="va">gs_ovnet</span>, markGroups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="va">g</span><span class="op">]</span>, genesetStat <span class="op">=</span> <span class="va">gsStats</span><span class="op">)</span>
<span class="va">p2_g</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigWordcloud.html">plotMsigWordcloud</a></span><span class="op">(</span>msigGsc <span class="op">=</span> <span class="va">siggs</span>, groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="va">g</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">'Short'</span><span class="op">)</span>
<span class="va">p3_g</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotGeneStats.html">plotGeneStats</a></span><span class="op">(</span>
  <span class="va">gStats</span>,
  msigGsc <span class="op">=</span> <span class="va">siggs</span>,
  groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="va">g</span><span class="op">]</span>,
  statName <span class="op">=</span> <span class="st">'logFC'</span>,
  topN <span class="op">=</span> <span class="fl">25</span>
<span class="op">)</span> <span class="op">+</span> <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, lty <span class="op">=</span> <span class="fl">2</span>, colour <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="va">p2_g</span> <span class="op">+</span> <span class="va">p1_g</span> <span class="op">+</span> <span class="va">p3_g</span></code></pre></div>
<p><img src="workflow_singscore_vissE_files/figure-html/unnamed-chunk-36-1.png" width="1440"></p>
</div>
</div>
<div id="summary" class="section level1">
<h1 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h1>
</div>
<div id="packages-used" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#packages-used" class="anchor"></a>Packages used</h1>
<p>This workflow depends on various packages from version 3.14 of the Bioconductor project, running on R version 4.1.0 (2021-05-18) or higher. The complete list of the packages used for this workflow are shown below:</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] igraph_1.2.6                  patchwork_1.1.1              
##  [3] vissE_1.1.0                   msigdb_1.1.0                 
##  [5] singscore_1.13.1              emtdata_1.1.0                
##  [7] GSEABase_1.55.1               graph_1.71.2                 
##  [9] annotate_1.71.0               XML_3.99-0.7                 
## [11] AnnotationDbi_1.55.1          edgeR_3.35.0                 
## [13] limma_3.49.4                  ExperimentHub_2.1.4          
## [15] AnnotationHub_3.1.5           BiocFileCache_2.1.1          
## [17] dbplyr_2.1.1                  SummarizedExperiment_1.23.1  
## [19] Biobase_2.53.0                GenomicRanges_1.45.0         
## [21] GenomeInfoDb_1.29.3           IRanges_2.27.2               
## [23] S4Vectors_0.31.1              BiocGenerics_0.39.2          
## [25] MatrixGenerics_1.5.3          matrixStats_0.60.1           
## [27] ggplot2_3.3.5                 GenesetAnalysisWorkflow_0.9.1
## 
## loaded via a namespace (and not attached):
##   [1] systemfonts_1.0.2             plyr_1.8.6                   
##   [3] splines_4.1.0                 digest_0.6.27                
##   [5] htmltools_0.5.1.1             fansi_0.5.0                  
##   [7] magrittr_2.0.1                memoise_2.0.0                
##   [9] tm_0.7-8                      Biostrings_2.61.2            
##  [11] textshape_1.7.3               pkgdown_1.6.1                
##  [13] colorspace_2.0-2              blob_1.2.2                   
##  [15] rappdirs_0.3.3                ggrepel_0.9.1                
##  [17] sylly_0.1-6                   textshaping_0.3.5            
##  [19] xfun_0.25                     dplyr_1.0.7                  
##  [21] crayon_1.4.1                  RCurl_1.98-1.4               
##  [23] jsonlite_1.7.2                hexbin_1.28.2                
##  [25] glue_1.4.2                    polyclip_1.10-0              
##  [27] gtable_0.3.0                  zlibbioc_1.39.0              
##  [29] XVector_0.33.0                DelayedArray_0.19.1          
##  [31] scico_1.2.0                   scales_1.1.1                 
##  [33] qdapRegex_0.7.2               DBI_1.1.1                    
##  [35] Rcpp_1.0.7                    viridisLite_0.4.0            
##  [37] xtable_1.8-4                  bit_4.0.4                    
##  [39] textclean_0.9.3               httr_1.4.2                   
##  [41] ggwordcloud_0.5.0             RColorBrewer_1.1-2           
##  [43] ellipsis_0.3.2                pkgconfig_2.0.3              
##  [45] farver_2.1.0                  sass_0.4.0                   
##  [47] locfit_1.5-9.4                utf8_1.2.2                   
##  [49] tidyselect_1.1.1              labeling_0.4.2               
##  [51] rlang_0.4.11                  reshape2_1.4.4               
##  [53] later_1.3.0                   munsell_0.5.0                
##  [55] BiocVersion_3.14.0            tools_4.1.0                  
##  [57] cachem_1.0.6                  generics_0.1.0               
##  [59] RSQLite_2.2.8                 evaluate_0.14                
##  [61] stringr_1.4.0                 fastmap_1.1.0                
##  [63] yaml_2.2.1                    ragg_1.1.3                   
##  [65] textstem_0.1.4                org.Hs.eg.db_3.13.0          
##  [67] knitr_1.33                    bit64_4.0.5                  
##  [69] fs_1.5.0                      purrr_0.3.4                  
##  [71] KEGGREST_1.33.0               koRpus_0.13-8                
##  [73] mime_0.11                     slam_0.1-48                  
##  [75] xml2_1.3.2                    compiler_4.1.0               
##  [77] filelock_1.0.2                curl_4.3.2                   
##  [79] png_0.1-7                     interactiveDisplayBase_1.31.2
##  [81] koRpus.lang.en_0.1-4          syuzhet_1.0.6                
##  [83] tibble_3.1.3                  statmod_1.4.36               
##  [85] tweenr_1.0.2                  bslib_0.2.5.1                
##  [87] stringi_1.7.3                 highr_0.9                    
##  [89] desc_1.3.0                    lattice_0.20-44              
##  [91] Matrix_1.3-4                  vctrs_0.3.8                  
##  [93] pillar_1.6.2                  lifecycle_1.0.0              
##  [95] BiocManager_1.30.16           jquerylib_0.1.4              
##  [97] data.table_1.14.0             bitops_1.0-7                 
##  [99] httpuv_1.6.2                  sylly.en_0.1-3               
## [101] R6_2.5.1                      promises_1.2.0.1             
## [103] KernSmooth_2.23-20            lexicon_1.2.1                
## [105] MASS_7.3-54                   assertthat_0.2.1             
## [107] rprojroot_2.0.2               withr_2.4.2                  
## [109] GenomeInfoDbData_1.2.6        parallel_4.1.0               
## [111] grid_4.1.0                    prettydoc_0.4.1              
## [113] tidyr_1.1.3                   rmarkdown_2.10               
## [115] ggnewscale_0.4.5              ggforce_0.3.3                
## [117] NLP_0.2-1                     shiny_1.6.0</code></pre>
</div>
<div id="acknowledgments" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#acknowledgments" class="anchor"></a>Acknowledgments</h1>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-R-rmarkdown" class="csl-entry">
Allaire, JJ, Yihui Xie, Jonathan McPherson, Javier Luraschi, Kevin Ushey, Aron Atkins, Hadley Wickham, Joe Cheng, Winston Chang, and Richard Iannone. 2021. <em>Rmarkdown: Dynamic Documents for r</em>. <a href="https://CRAN.R-project.org/package=rmarkdown">https://CRAN.R-project.org/package=rmarkdown</a>.
</div>
<div id="ref-bhuva20" class="csl-entry">
Bhuva, Dharmesh D, Joseph Cursons, and Melissa J Davis. 2020. <span>“Stable Gene Expression for Normalisation and Single-Sample Scoring.”</span> <em>Nucleic Acids Research</em> 48 (19): e113–13.
</div>
<div id="ref-bhuva19" class="csl-entry">
Bhuva, Dharmesh D, Momeneh Foroutan, Yi Xie, Ruqian Lyu, Joseph Cursons, and Melissa J Davis. 2019. <span>“Using Singscore to Predict Mutation Status in Acute Myeloid Leukemia from Transcriptomic Signatures.”</span> <em>F1000Research</em> 8.
</div>
<div id="ref-R-edgeR" class="csl-entry">
Chen, Yunshun, Aaron TL Lun, Davis J McCarthy, Matthew E Ritchie, Belinda Phipson, Yifang Hu, Xiaobei Zhou, Mark D Robinson, and Gordon K Smyth. 2021. <em>edgeR: Empirical Analysis of Digital Gene Expression Data in r</em>.
</div>
<div id="ref-chen16" class="csl-entry">
Chen, Yunshun, Aaron TL Lun, and Gordon K Smyth. 2016. <span>“From Reads to Genes to Pathways: Differential Expression Analysis of RNA-Seq Experiments Using Rsubread and the edgeR Quasi-Likelihood Pipeline.”</span> <em>F1000Research</em> 5.
</div>
<div id="ref-cursons15" class="csl-entry">
Cursons, Joseph, Karl-Johan Leuchowius, Mark Waltham, Eva Tomaskovic-Crook, Momeneh Foroutan, Cameron P Bracken, Andrew Redfern, et al. 2015. <span>“Stimulus-Dependent Differences in Signalling Regulate Epithelial-Mesenchymal Plasticity and Change the Effects of Drugs in Breast Cancer Cell Lines.”</span> <em>Cell Communication and Signaling</em> 13 (1): 1–21.
</div>
<div id="ref-cursons18" class="csl-entry">
Cursons, Joseph, Katherine A Pillman, Kaitlin G Scheer, Philip A Gregory, Momeneh Foroutan, Soroor Hediyeh-Zadeh, John Toubia, et al. 2018. <span>“Combinatorial Targeting by microRNAs Co-Ordinates Post-Transcriptional Control of EMT.”</span> <em>Cell Systems</em> 7 (1): 77–91.
</div>
<div id="ref-foroutan18" class="csl-entry">
Foroutan, Momeneh, Dharmesh D Bhuva, Ruqian Lyu, Kristy Horan, Joseph Cursons, and Melissa J Davis. 2018. <span>“Single Sample Scoring of Molecular Phenotypes.”</span> <em>BMC Bioinformatics</em> 19 (1): 404.
</div>
<div id="ref-foroutan17" class="csl-entry">
Foroutan, Momeneh, Joseph Cursons, Soroor Hediyeh-Zadeh, Erik W Thompson, and Melissa J Davis. 2017. <span>“A Transcriptional Program for Detecting TGF<span class="math inline">\(\beta\)</span>-Induced EMT in Cancer.”</span> <em>Molecular Cancer Research</em> 15 (5): 619–31.
</div>
<div id="ref-liao13" class="csl-entry">
Liao, Yang, Gordon K Smyth, and Wei Shi. 2013. <span>“The Subread Aligner: Fast, Accurate and Scalable Read Mapping by Seed-and-Vote.”</span> <em>Nucleic Acids Research</em> 41 (10): e108–8.
</div>
<div id="ref-liao14" class="csl-entry">
———. 2014. <span>“featureCounts: An Efficient General Purpose Program for Assigning Sequence Reads to Genomic Features.”</span> <em>Bioinformatics</em> 30 (7): 923–30.
</div>
<div id="ref-liberzon15" class="csl-entry">
Liberzon, Arthur, Chet Birger, Helga Thorvaldsdóttir, Mahmoud Ghandi, Jill P Mesirov, and Pablo Tamayo. 2015. <span>“The Molecular Signatures Database Hallmark Gene Set Collection.”</span> <em>Cell Systems</em> 1 (6): 417–25.
</div>
<div id="ref-R-prettydoc" class="csl-entry">
Qiu, Yixuan. 2021. <em>Prettydoc: Creating Pretty Documents from r Markdown</em>. <a href="https://github.com/yixuan/prettydoc">https://github.com/yixuan/prettydoc</a>.
</div>
<div id="ref-robinson10" class="csl-entry">
Robinson, Mark D, and Alicia Oshlack. 2010. <span>“A Scaling Normalization Method for Differential Expression Analysis of RNA-Seq Data.”</span> <em>Genome Biology</em> 11 (3): 1–9.
</div>
<div id="ref-subramanian05" class="csl-entry">
Subramanian, Aravind, Pablo Tamayo, Vamsi K Mootha, Sayan Mukherjee, Benjamin L Ebert, Michael A Gillette, Amanda Paulovich, et al. 2005. <span>“Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.”</span> <em>Proceedings of the National Academy of Sciences</em> 102 (43): 15545–50.
</div>
<div id="ref-tan14" class="csl-entry">
Tan, Tuan Zea, Qing Hao Miow, Yoshio Miki, Tetsuo Noda, Seiichi Mori, Ruby Yun-Ju Huang, and Jean Paul Thiery. 2014. <span>“Epithelial-Mesenchymal Transition Spectrum Quantification and Its Efficacy in Deciphering Survival and Drug Responses of Cancer Patients.”</span> <em>EMBO Molecular Medicine</em> 6 (10): 1279–93.
</div>
<div id="ref-R-ggplot2" class="csl-entry">
Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen, Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, and Dewey Dunnington. 2021. <em>Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics</em>. <a href="https://CRAN.R-project.org/package=ggplot2">https://CRAN.R-project.org/package=ggplot2</a>.
</div>
<div id="ref-R-knitr" class="csl-entry">
Xie, Yihui. 2021. <em>Knitr: A General-Purpose Package for Dynamic Report Generation in r</em>. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Dharmesh D. Bhuva, Chin Wee Tan, Melissa J. Davis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
