<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>使用R包singscore，vissE和msigdb对单样本进行分子表型可视化 • GenesetAnalysisWorkflow</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="使用R包singscore，vissE和msigdb对单样本进行分子表型可视化">
<meta property="og:description" content="GenesetAnalysisWorkflow">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenesetAnalysisWorkflow</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/workflow_singscore_vissE.html">Visualising molecular phenotypes inferred from individual samples using singscore, vissE and msigdb</a>
    </li>
    <li>
      <a href="../articles/workflow_singscore_vissE_Chinese_ver.html">使用R包singscore，vissE和msigdb对单样本进行分子表型可视化</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/DavisLaboratory/GenesetAnalysisWorkflow">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workflow_singscore_vissE_Chinese_ver_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>使用R包singscore，vissE和msigdb对单样本进行分子表型可视化</h1>
                        <h4 class="author">Dharmesh D Bhuva</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of Medical Research, Parkville, VIC 3052, AustraliaDepartment of Medical Biology, University of Melbourne, Parkville, VIC 3010, Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:bhuva.d@wehi.edu.au" class="email">bhuva.d@wehi.edu.au</a>
      </address>
                              <h4 class="author">Chin Wee Tan</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of Medical Research, Parkville, VIC 3052, AustraliaDepartment of Medical Biology, University of Melbourne, Parkville, VIC 3010, Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:cwtan@wehi.edu.au" class="email">cwtan@wehi.edu.au</a>
      </address>
                              <h4 class="author">Jinjin Chen</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of Medical Research, Parkville, VIC 3052, AustraliaDepartment of Medical Biology, University of Melbourne, Parkville, VIC 3010, Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:chen.j@wehi.edu.au" class="email">chen.j@wehi.edu.au</a>
      </address>
                              <h4 class="author">Melissa J Davis</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of Medical Research, Parkville, VIC 3052, AustraliaDepartment of Medical Biology, University of Melbourne, Parkville, VIC 3010, AustraliaDepartment of Biochemistry and Molecular Biology, Faculty of Medicine, Dentistry and Health Sciences, University of Melbourne, Parkville, VIC, 3010, Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:davis.m@wehi.edu.au" class="email">davis.m@wehi.edu.au</a>
      </address>
                  
            <h4 class="date">Oct 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/GenesetAnalysisWorkflow/blob/master/vignettes/workflow_singscore_vissE_Chinese_ver.Rmd"><code>vignettes/workflow_singscore_vissE_Chinese_ver.Rmd</code></a></small>
      <div class="hidden name"><code>workflow_singscore_vissE_Chinese_ver.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>摘要</p>
    </div>
    
<p>
</p>
<p><strong>R version</strong>: R version 4.1.1 (2021-08-10) <br><strong>Bioconductor version</strong>: 3.14 <br><strong>Package version</strong>: 0.9.2</p>

<div id="背景介绍" class="section level1">
<h1 class="hasAnchor">
<a href="#%E8%83%8C%E6%99%AF%E4%BB%8B%E7%BB%8D" class="anchor"></a>背景介绍</h1>
<p>一套标准的生信组学数据分析流程会通过统计学分析产生一个分子列表。在转录组学背景中，这些分子往往是基因或者转录本，而鉴定它们的统计学方法通常是差异表达分析（DE分析）。一旦基因在实验中被鉴定为差异表达基因，生物学家们常常会想了解它们的生物学意义。而这是通过了解它们在所研究的生物学系统中的功能作用来实现的。许多基因的角色和功能在某种程度上是已知的，这也是一项持续进行的研究。基因功能的有关知识通常被编码在知识库中，如基因本体（GO）以及其他的通路数据库。基于这些功能注释，我们有兴趣在我们的数据中鉴定那些被富集的功能。</p>
<p>为此，我们使用基因集富集分析，这是一组用于鉴定基因集中被富集的功能的方法。这个分析流程会展示如何使用分子标志数据库（通过<code>msigdb</code> R/Bioconductor包）和基因集富集方法（<code>singscore</code>），对转录组数据进行功能分析。它还将展示如何使用<code>vissE</code>包在数据中鉴定更高阶的生物学主题。本分析流程将使用<code>emtdata</code>和<code>msigdb</code> R/Bioconductor包从<code>ExperimentHub</code>中加载基因表达数据以及相关基因集。单样本基因集富集方法（<code>singscore</code>）将被用来鉴定表征了样本功能特征的分子表型。最后，使用<code>vissE</code>对更高阶的功能主题进行鉴定。</p>
</div>
<div id="生物学问题的描述" class="section level1">
<h1 class="hasAnchor">
<a href="#%E7%94%9F%E7%89%A9%E5%AD%A6%E9%97%AE%E9%A2%98%E7%9A%84%E6%8F%8F%E8%BF%B0" class="anchor"></a>生物学问题的描述</h1>
<p>上皮-间充质转化（EMT）是一种细胞过程，描述了静态极化的上皮细胞转化为活动的间充质细胞表型的现象。它是涉及胚胎发生和组织分化的基础机制。而在癌症方面，它与癌症进展和转移有关。EMT导致上皮特征的逐渐丧失以及间充质特征的获得，使得癌症中的细胞获得了转移与侵袭的特性，诱导了细胞干性，从而避免了细胞凋亡和衰老，具备了免疫抗性。(<span class="citation">Ksiazkiewicz, Markiewicz, and Zaczek (2012)</span>)</p>
<p>EMT的特征是上皮标志物（如E-钙粘蛋白和密封蛋白）的下调以及间充质标志物（如波形蛋白和N-钙粘蛋白）的上调。这些标志物的改变会导致细胞间的粘附力和细胞极性的丧失，以及迁移性和侵袭性的获得。而从富含角蛋白的粘连网络转换成富含波形蛋白的黏着斑网络则伴随着钙粘蛋白从E型转为N型<span class="citation">(Tomita et al. 2000)</span>。</p>
<p>尽管癌细胞通过EMT促进转移扩散，但它们很少完全转化为间充质表型，而是表现出一种杂交表型，被宽泛地称为上皮间充质可塑性（EMP）。EMP与乳腺癌转移风险的增加以及不良预后相关<span class="citation">(Cursons et al. 2015)</span>。受限于缺乏可靠的进展读数以及由此导致的对EMT的过渡与进展状态的有限研究，目前针对这种瞬态/亚稳态表型提出的特定标志物有限。这种混合态的特定标志物的鉴定将为确定转移的干预措施，以及揭示潜在治疗方法的机制提供关键信息。<span class="citation">(Ribeiro and Paredes 2015)</span></p>
<p>在本次分析流程中，我们使用来自<span class="citation">(Cursons et al. 2015)</span>研究的批量转录组数据，该研究在两种乳腺癌细胞系模型中，即PMC42 system（PMC42-ET和PMC42-LA亚株）和MDA-MB-468，对EMP进行研究。在该数据集中，PMC42-LA是一种源自波形蛋白阳性，E-钙粘蛋白阴性的亲本PMC42-ET细胞的上皮亚系，两者均被归类为具有”基底B”转录组（表达间充质基因，缺乏上皮标记基因表达）。</p>
</div>
<div id="单个样本的分子表型" class="section level1">
<h1 class="hasAnchor">
<a href="#%E5%8D%95%E4%B8%AA%E6%A0%B7%E6%9C%AC%E7%9A%84%E5%88%86%E5%AD%90%E8%A1%A8%E5%9E%8B" class="anchor"></a>单个样本的分子表型</h1>
<p>为了对来自<span class="citation">(Cursons et al. 2015)</span>的数据进行分子表型分析，我们需要RNA-seq计数以及能表征感兴趣的分子表型的合适的基因集。在本次分析流程中我们将使用来自<code>emtdata</code>包的处理过的RNA-seq数据。接下来我们将使用singscore方法(<span class="citation">Bhuva, Cursons, and Davis (2020)</span>)，分别针对来自<span class="citation">(Tan et al. 2014)</span>和分子标志数据库（MSigDB）(<span class="citation">Subramanian et al. (2005)</span>)的自定义基因标志物对单个样本进行评分。</p>
<div id="加载数据" class="section level2">
<h2 class="hasAnchor">
<a href="#%E5%8A%A0%E8%BD%BD%E6%95%B0%E6%8D%AE" class="anchor"></a>加载数据</h2>
<p><code>emtdata</code>包中包含了多种预处理过的RNA-seq数据，这些数据来自研究上皮向间充质转化的实验。该包包含了以下经过处理的数据集：</p>
<ol style="list-style-type: decimal">
<li>cursons2015_se - 来自多种乳腺癌细胞系的RNA-seq数据，代表不同程度的上皮和间充质表型<span class="citation">(Cursons et al. 2015)</span>。</li>
<li>cursons2018_se - 来自人乳腺上皮细胞（HMLE）系的RNA-seq数据，该细胞系经TGF<span class="math inline">\(\beta\)</span>处理诱导上皮间充质转化（EMT），随后以miR-200c处理诱导间充质上皮转化（MET）<span class="citation">(Cursons et al. 2018)</span>.</li>
<li>foroutan2017_se - 一份EMT诱导实验的汇编数据，结合并消除了批次效应<span class="citation">(Foroutan et al. 2017)</span>.</li>
</ol>
<p>本次分析流程中使用的数据为<span class="citation">(Cursons et al. 2015)</span>RNA-seq数据。从SRA下载RNA-seq数据，并使用Subread aligner<span class="citation">(Liao, Smyth, and Shi 2013)</span>和featureCounts<span class="citation">(Liao, Smyth, and Shi 2014)</span>对数据进行处理生成计数矩阵。随后使用<code><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html">edgeR::filterByExpr()</a></code>函数过滤掉低表达基因。使用<code>edgeR</code> R/Bioconductor包中的相关函数，根据TMM计算得到标准化因子，并计算基因的每百万读数中每千碱基映射读数（RPKM）。随后数据被存储在SummarizedExperiment对象中，可以使用<code>emtdata</code> R/Bioconductor包从<code>ExperimentHub</code>中被检索到。</p>
<p>来自<code>emtdata</code>包的数据可以使用包内的访问器或者<code>ExperimentHub</code>被检索到。以下的检索将搜索hub中与搜索词”emtdata”相关联的所有对象。检索到<code>emtdata</code>包中可用的三个数据集的信息。</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span>

<span class="va">eh</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">res</span> <span class="op">=</span> <span class="fu">query</span><span class="op">(</span><span class="va">eh</span>, <span class="st">'emtdata'</span><span class="op">)</span>
<span class="va">res</span></code></pre></div>
<pre><code>## ExperimentHub with 3 records
## # snapshotDate(): 2021-10-18
## # $dataprovider: Walter and Eliza Hall Institute of Medical Research, Queens...
## # $species: Homo sapiens
## # $rdataclass: GSEABase::SummarizedExperiment
## # additional mcols(): taxonomyid, genome, description,
## #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,
## #   rdatapath, sourceurl, sourcetype 
## # retrieve records with, e.g., 'object[["EH5439"]]' 
## 
##            title          
##   EH5439 | foroutan2017_se
##   EH5440 | cursons2018_se 
##   EH5441 | cursons2015_se</code></pre>
<p>我们可以使用<code>mcols()</code>函数查看与每个对象关联的所有元数据。该信息将链接到每个数据的原始出版物，我们还能查看产生这些数据的物种信息。</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#检索元数据</span>
<span class="fu">mcols</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 3 rows and 15 columns
##                  title           dataprovider      species taxonomyid
##            &lt;character&gt;            &lt;character&gt;  &lt;character&gt;  &lt;integer&gt;
## EH5439 foroutan2017_se Walter and Eliza Hal.. Homo sapiens       9606
## EH5440  cursons2018_se Queensland Universit.. Homo sapiens       9606
## EH5441  cursons2015_se Center for Cancer Bi.. Homo sapiens       9606
##             genome            description coordinate_1_based
##        &lt;character&gt;            &lt;character&gt;          &lt;integer&gt;
## EH5439          NA Gene expression data..                  1
## EH5440          NA Gene expression data..                  1
## EH5441          NA Gene expression data..                  1
##                    maintainer rdatadateadded preparerclass
##                   &lt;character&gt;    &lt;character&gt;   &lt;character&gt;
## EH5439 Malvika D. Kharbanda..     2021-03-30       emtdata
## EH5440 Malvika D. Kharbanda..     2021-03-30       emtdata
## EH5441 Malvika D. Kharbanda..     2021-03-30       emtdata
##                                             tags             rdataclass
##                                           &lt;list&gt;            &lt;character&gt;
## EH5439         Gene_Expression,Homo_sapiens_Data GSEABase::Summarized..
## EH5440                    HMLE,Homo_sapiens_Data GSEABase::Summarized..
## EH5441 Homo_sapiens_Data,MDA-MB-468,PMC42-ET,... GSEABase::Summarized..
##                     rdatapath              sourceurl  sourcetype
##                   &lt;character&gt;            &lt;character&gt; &lt;character&gt;
## EH5439 emtdata/foroutan2017.. https://doi.org/10.4..         TXT
## EH5440 emtdata/cursons2018_.. https://www.ncbi.nlm..         TXT
## EH5441 emtdata/cursons2015_.. https://www.ncbi.nlm..         TXT</code></pre>
<p>可以使用编号（“EH5441”）或者通过函数<code><a href="https://rdrr.io/pkg/emtdata/man/cursons2015_se.html">cursons2015_se()</a></code>来检索对象。数据被存储为<code>SummarizedExperiment</code>对象，函数<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData()</a></code>，<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData()</a></code>和<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay()</a></code>可用于与对象交互。我们能看见该数据测量了21个样本中约30,000个基因，基因的计数和logRPKM矩阵都被存储在对象中。</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/DavisLaboratory/emtdata">emtdata</a></span><span class="op">)</span>

<span class="co">#使用对象编号加载数据</span>
<span class="va">emt_se</span> <span class="op">=</span> <span class="va">eh</span><span class="op">[[</span><span class="st">'EH5441'</span><span class="op">]</span><span class="op">]</span>

<span class="co">#等效方法</span>
<span class="va">emt_se</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/emtdata/man/cursons2015_se.html">cursons2015_se</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">emt_se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 29866 21 
## metadata(0):
## assays(2): counts logRPKM
## rownames(29866): ENSG00000223972 ENSG00000227232 ... ENSG00000271254
##   ENSG00000275405
## rowData names(7): Chr Start ... gene_name gene_biotype
## colnames(21): MDA468_Ctrl_Rep1 MDA468_Ctrl_Rep2 ... PMC42LA_EGF_Rep2
##   PMC42LA_EGF_Rep3
## colData names(14): group lib.size ... Organism SRA.Study</code></pre>
<p>样本注释显示该数据中有3种细胞系（PMC42-ET，PMC42-LA和MDA-MB-648）。这些细胞系经过缺氧和EGF处理，并在处理3或7天后进行测量。我们还获取了每个样本关联的元数据，包括原始SRA编号以及它们的文库大小。</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#查看样本注释</span>
<span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">emt_se</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##                   group  lib.size norm.factors                              Run
## MDA468_Ctrl_Rep1      1 114794500    0.9743731 SRR3576739,SRR3576768,SRR3591405
## MDA468_Ctrl_Rep2      1 114491243    0.9931838 SRR3576740,SRR3576770,SRR3591406
## MDA468_Ctrl_Rep3      1 107923317    1.0358755 SRR3576751,SRR3576784,SRR3591417
## MDA468_EGF_Rep1       1 120949415    0.9793903 SRR3576753,SRR3576786,SRR3591419
## MDA468_EGF_Rep2       1 108830760    0.9865599 SRR3576754,SRR3576788,SRR3591420
## MDA468_EGF_Rep3       1  95608850    0.9006452 SRR3576755,SRR3576790,SRR3591421
## MDA468_HPX_Rep1       1 103636210    0.9782668 SRR3576756,SRR3576791,SRR3591422
## MDA468_HPX_Rep2       1  99763188    1.0218358 SRR3576757,SRR3576792,SRR3591423
## MDA468_HPX_Rep3       1 111820864    0.9914170 SRR3576758,SRR3576793,SRR3591424
## PMC42ET_Ctrl_Rep1     1  71276286    1.0443833 SRR3576759,SRR3576795,SRR3591425
## PMC42ET_Ctrl_Rep2     1  74625943    0.9957722 SRR3576741,SRR3576771,SRR3591407
## PMC42ET_Ctrl_Rep3     1 104167668    1.0622063 SRR3576742,SRR3576772,SRR3591408
## PMC42ET_EGF_Rep1      1  74711695    1.0303307 SRR3576743,SRR3576773,SRR3591409
## PMC42ET_EGF_Rep2      1  87532741    1.0312069 SRR3576744,SRR3576775,SRR3591410
## PMC42ET_EGF_Rep3      1 102691326    1.0485896 SRR3576745,SRR3576776,SRR3591411
## PMC42LA_Ctrl_Rep1     1  66771046    0.9511911 SRR3576746,SRR3576777,SRR3591412
## PMC42LA_Ctrl_Rep2     1  89935383    0.9872186 SRR3576747,SRR3576779,SRR3591413
## PMC42LA_Ctrl_Rep3     1  74822017    0.9930820 SRR3576748,SRR3576781,SRR3591414
## PMC42LA_EGF_Rep1      1  80151428    0.9440959 SRR3576749,SRR3576782,SRR3591415
## PMC42LA_EGF_Rep2      1  88233254    1.0227524 SRR3576750,SRR3576783,SRR3591416
## PMC42LA_EGF_Rep3      1  85941564    1.0437588 SRR3576752,SRR3576785,SRR3591418
##                         Sample.Name    Subline    Treatment  BioProject
## MDA468_Ctrl_Rep1   MDA468_Ctrl_Rep1 MDA-MB-468      control PRJNA322427
## MDA468_Ctrl_Rep2   MDA468_Ctrl_Rep2 MDA-MB-468      control PRJNA322427
## MDA468_Ctrl_Rep3   MDA468_Ctrl_Rep3 MDA-MB-468      control PRJNA322427
## MDA468_EGF_Rep1     MDA468_EGF_Rep1 MDA-MB-468     EGF - 7d PRJNA322427
## MDA468_EGF_Rep2     MDA468_EGF_Rep2 MDA-MB-468     EGF - 7d PRJNA322427
## MDA468_EGF_Rep3     MDA468_EGF_Rep3 MDA-MB-468     EGF - 7d PRJNA322427
## MDA468_HPX_Rep1     MDA468_HPX_Rep1 MDA-MB-468 Hypoxia - 7d PRJNA322427
## MDA468_HPX_Rep2     MDA468_HPX_Rep2 MDA-MB-468 Hypoxia - 7d PRJNA322427
## MDA468_HPX_Rep3     MDA468_HPX_Rep3 MDA-MB-468 Hypoxia - 7d PRJNA322427
## PMC42ET_Ctrl_Rep1 PMC42ET_Ctrl_Rep1   PMC42-ET      control PRJNA322427
## PMC42ET_Ctrl_Rep2 PMC42ET_Ctrl_Rep2   PMC42-ET      control PRJNA322427
## PMC42ET_Ctrl_Rep3 PMC42ET_Ctrl_Rep3   PMC42-ET      control PRJNA322427
## PMC42ET_EGF_Rep1   PMC42ET_EGF_Rep1   PMC42-ET     EGF - 3d PRJNA322427
## PMC42ET_EGF_Rep2   PMC42ET_EGF_Rep2   PMC42-ET     EGF - 3d PRJNA322427
## PMC42ET_EGF_Rep3   PMC42ET_EGF_Rep3   PMC42-ET     EGF - 3d PRJNA322427
## PMC42LA_Ctrl_Rep1 PMC42LA_Ctrl_Rep1   PMC42-LA      control PRJNA322427
## PMC42LA_Ctrl_Rep2 PMC42LA_Ctrl_Rep2   PMC42-LA      control PRJNA322427
## PMC42LA_Ctrl_Rep3 PMC42LA_Ctrl_Rep3   PMC42-LA      control PRJNA322427
## PMC42LA_EGF_Rep1   PMC42LA_EGF_Rep1   PMC42-LA     EGF - 3d PRJNA322427
## PMC42LA_EGF_Rep2   PMC42LA_EGF_Rep2   PMC42-LA     EGF - 3d PRJNA322427
## PMC42LA_EGF_Rep3   PMC42LA_EGF_Rep3   PMC42-LA     EGF - 3d PRJNA322427
##                      BioSample                         Center.Name
## MDA468_Ctrl_Rep1  SAMN05162532 Queensland University of Technology
## MDA468_Ctrl_Rep2  SAMN05162533 Queensland University of Technology
## MDA468_Ctrl_Rep3  SAMN05162534 Queensland University of Technology
## MDA468_EGF_Rep1   SAMN05162535 Queensland University of Technology
## MDA468_EGF_Rep2   SAMN05162536 Queensland University of Technology
## MDA468_EGF_Rep3   SAMN05162537 Queensland University of Technology
## MDA468_HPX_Rep1   SAMN05162538 Queensland University of Technology
## MDA468_HPX_Rep2   SAMN05162539 Queensland University of Technology
## MDA468_HPX_Rep3   SAMN05162540 Queensland University of Technology
## PMC42ET_Ctrl_Rep1 SAMN05162541 Queensland University of Technology
## PMC42ET_Ctrl_Rep2 SAMN05162542 Queensland University of Technology
## PMC42ET_Ctrl_Rep3 SAMN05162543 Queensland University of Technology
## PMC42ET_EGF_Rep1  SAMN05162544 Queensland University of Technology
## PMC42ET_EGF_Rep2  SAMN05162545 Queensland University of Technology
## PMC42ET_EGF_Rep3  SAMN05162546 Queensland University of Technology
## PMC42LA_Ctrl_Rep1 SAMN05162547 Queensland University of Technology
## PMC42LA_Ctrl_Rep2 SAMN05162548 Queensland University of Technology
## PMC42LA_Ctrl_Rep3 SAMN05162549 Queensland University of Technology
## PMC42LA_EGF_Rep1  SAMN05162550 Queensland University of Technology
## PMC42LA_EGF_Rep2  SAMN05162551 Queensland University of Technology
## PMC42LA_EGF_Rep3  SAMN05162552 Queensland University of Technology
##                                         Experiment  Cell.Line     Organism
## MDA468_Ctrl_Rep1  SRX1795037,SRX1795064,SRX1802009 MDA-MB-468 Homo sapiens
## MDA468_Ctrl_Rep2  SRX1795038,SRX1795066,SRX1802010 MDA-MB-468 Homo sapiens
## MDA468_Ctrl_Rep3  SRX1795049,SRX1795078,SRX1802021 MDA-MB-468 Homo sapiens
## MDA468_EGF_Rep1   SRX1795051,SRX1795080,SRX1802023 MDA-MB-468 Homo sapiens
## MDA468_EGF_Rep2   SRX1795052,SRX1795081,SRX1802024 MDA-MB-468 Homo sapiens
## MDA468_EGF_Rep3   SRX1795053,SRX1795082,SRX1802025 MDA-MB-468 Homo sapiens
## MDA468_HPX_Rep1   SRX1795054,SRX1795083,SRX1802026 MDA-MB-468 Homo sapiens
## MDA468_HPX_Rep2   SRX1795055,SRX1795084,SRX1802027 MDA-MB-468 Homo sapiens
## MDA468_HPX_Rep3   SRX1795056,SRX1795086,SRX1802028 MDA-MB-468 Homo sapiens
## PMC42ET_Ctrl_Rep1 SRX1795057,SRX1795087,SRX1802029   PMC42-ET Homo sapiens
## PMC42ET_Ctrl_Rep2 SRX1795039,SRX1795067,SRX1802011   PMC42-ET Homo sapiens
## PMC42ET_Ctrl_Rep3 SRX1795040,SRX1795068,SRX1802012   PMC42-ET Homo sapiens
## PMC42ET_EGF_Rep1  SRX1795041,SRX1795069,SRX1802013   PMC42-ET Homo sapiens
## PMC42ET_EGF_Rep2  SRX1795042,SRX1795070,SRX1802014   PMC42-ET Homo sapiens
## PMC42ET_EGF_Rep3  SRX1795043,SRX1795071,SRX1802015   PMC42-ET Homo sapiens
## PMC42LA_Ctrl_Rep1 SRX1795044,SRX1795072,SRX1802016   PMC42-LA Homo sapiens
## PMC42LA_Ctrl_Rep2 SRX1795045,SRX1795073,SRX1802017   PMC42-LA Homo sapiens
## PMC42LA_Ctrl_Rep3 SRX1795046,SRX1795074,SRX1802018   PMC42-LA Homo sapiens
## PMC42LA_EGF_Rep1  SRX1795047,SRX1795075,SRX1802019   PMC42-LA Homo sapiens
## PMC42LA_EGF_Rep2  SRX1795048,SRX1795076,SRX1802020   PMC42-LA Homo sapiens
## PMC42LA_EGF_Rep3  SRX1795050,SRX1795079,SRX1802022   PMC42-LA Homo sapiens
##                   SRA.Study
## MDA468_Ctrl_Rep1  SRP075592
## MDA468_Ctrl_Rep2  SRP075592
## MDA468_Ctrl_Rep3  SRP075592
## MDA468_EGF_Rep1   SRP075592
## MDA468_EGF_Rep2   SRP075592
## MDA468_EGF_Rep3   SRP075592
## MDA468_HPX_Rep1   SRP075592
## MDA468_HPX_Rep2   SRP075592
## MDA468_HPX_Rep3   SRP075592
## PMC42ET_Ctrl_Rep1 SRP075592
## PMC42ET_Ctrl_Rep2 SRP075592
## PMC42ET_Ctrl_Rep3 SRP075592
## PMC42ET_EGF_Rep1  SRP075592
## PMC42ET_EGF_Rep2  SRP075592
## PMC42ET_EGF_Rep3  SRP075592
## PMC42LA_Ctrl_Rep1 SRP075592
## PMC42LA_Ctrl_Rep2 SRP075592
## PMC42LA_Ctrl_Rep3 SRP075592
## PMC42LA_EGF_Rep1  SRP075592
## PMC42LA_EGF_Rep2  SRP075592
## PMC42LA_EGF_Rep3  SRP075592</code></pre>
<p>该数据使用的基因标识符是Ensembl基因标志符，但下游分析需要用到基因symbols和Entrez IDs。我们将把Ensembl IDs转换为基因symbols，因为这些更容易被直观地理解。<code>emt_se</code>的行注释包含了其他标识符的映射，如symbols。我们可以使用存储在对象中的映射将Ensembl IDs转化成symbols。但由于不同标识符间存在多重映射的问题，我们首先需要处理重复映射。我们可以通过以下方式识别重复项。</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#识别重复映射</span>
<span class="va">all_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">emt_se</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span>
<span class="va">dups</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="va">dups</span></code></pre></div>
<pre><code>##  [1] "Y_RNA"       "RGS5"        "TBCE"        "LINC00486"   "Metazoa_SRP"
##  [6] "LINC01238"   "CYB561D2"    "POLR2J4"     "POLR2J3"     "5S_rRNA"    
## [11] "TMSB15B"     "ALG1L9P"     "DNAJC9-AS1"  "BMS1P4"      "SNORA70"    
## [16] "HERC2P7"     "U2"          "U3"          "ELFN2"       "5_8S_rRNA"  
## [21] "U1"</code></pre>
<p>大部分重复项是由MsicRNA引起的，因此可以通过使用其他注释，例如biotypes（比如保留蛋白编码基因）来手动移除或解决。由于重复映射基因的数量相对较小，且这些基因与我们希望研究的过程（EMT）无关，因此舍弃它们的影响将很小，可以被安全忽略。为了简单起见，我们在本次分析中直接舍弃重复映射基因。</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#移除重复映射基因</span>
<span class="va">emt_se</span> <span class="op">=</span> <span class="va">emt_se</span><span class="op">[</span><span class="op">!</span><span class="va">all_genes</span> <span class="op">%in%</span> <span class="va">dups</span>, <span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">emt_se</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">emt_se</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span>
<span class="va">emt_se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 29752 21 
## metadata(0):
## assays(2): counts logRPKM
## rownames(29752): DDX11L1 WASH7P ... AC004556.3 AC240274.1
## rowData names(7): Chr Start ... gene_name gene_biotype
## colnames(21): MDA468_Ctrl_Rep1 MDA468_Ctrl_Rep2 ... PMC42LA_EGF_Rep2
##   PMC42LA_EGF_Rep3
## colData names(14): group lib.size ... Organism SRA.Study</code></pre>
</div>
<div id="加载基因集" class="section level2">
<h2 class="hasAnchor">
<a href="#%E5%8A%A0%E8%BD%BD%E5%9F%BA%E5%9B%A0%E9%9B%86" class="anchor"></a>加载基因集</h2>
<p>需要准备好表征感兴趣的分子表型的基因集用来查询单个样本。我们既可以使用<code>msigdb</code>包来获取分子标志数据库中的标志基因，也可以使用来自出版刊物发表的自定义标志基因。<span class="citation">Tan et al. (2014)</span> 开发了表征上皮和间充质表型的基因集，我们将使用这些基因集来探究样本的上皮-间充质结构。出版物中发表的数据被写入了<em>Thiery_EMTsignatures.txt</em>文件，该文件存储在与此分析流程相关的包中。</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#读取Theiry et al.的标志基因集</span>
<span class="va">thiery_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">'extdata/Thiery_EMTsignatures.txt'</span>, package <span class="op">=</span> <span class="st">'GenesetAnalysisWorkflow'</span><span class="op">)</span>
<span class="va">thiery_data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="va">thiery_path</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">thiery_data</span><span class="op">)</span></code></pre></div>
<pre><code>##   officialSymbol     genes epiMes_tumor epiMes_cellLine Ensembl.Gene.ID
## 1          ABCC3     ABCC3          epi             epi ENSG00000108846
## 2         ABHD11    ABHD11          epi            &lt;NA&gt; ENSG00000106077
## 3          ADAP1     ADAP1         &lt;NA&gt;             epi ENSG00000105963
## 4          ADIRF C10orf116         &lt;NA&gt;             epi ENSG00000148671
## 5           AGR2      AGR2          epi             epi ENSG00000106541
## 6           AIM1      AIM1         &lt;NA&gt;             epi ENSG00000112297
##   HGNC.symbol EntrezGene.ID
## 1       ABCC3          8714
## 2      ABHD11         83451
## 3       ADAP1         11033
## 4       ADIRF         10974
## 5        AGR2         10551
## 6        AIM1           202</code></pre>
<p>由于基因表达数据中使用的是基因symbols，我们将使用symbols创建基因集。我们将使用来自<span class="citation">(Tan et al. 2014)</span>细胞系的特异性标志基因，基因集被存储在专用的<code><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-methods.html">GSEABase::GeneSet</a></code>对象中，可以用于存储基因集相关的元数据。Singscore既可以使用被定义为字符串向量的基因集，也可以处理<code><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-methods.html">GSEABase::GeneSet</a></code>对象。由于后者具有更好的数据结构，我们倾向于在分析中使用后者。在后续的vissE分析中也使用到了GeneSet对象，更进一步推动了它们的使用。GeneSet对象要求基因具有唯一的标识符，因此最好为每一个基因集指定一个唯一且定义明确的集合名。此外，为了后续使用<code>vissE</code>包，我们将在此指定标识符类型（即<code><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html">SymbolIdentifier()</a></code>）。</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span>

<span class="co">#检索上皮基因</span>
<span class="va">epi_genes</span> <span class="op">=</span> <span class="va">thiery_data</span><span class="op">$</span><span class="va">officialSymbol</span><span class="op">[</span><span class="va">thiery_data</span><span class="op">$</span><span class="va">epiMes_cellLine</span> <span class="op">%in%</span> <span class="st">'epi'</span><span class="op">]</span>
<span class="co">#过滤重复基因</span>
<span class="va">epi_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">epi_genes</span><span class="op">)</span>
<span class="co">#创建GeneSet对象</span>
<span class="va">epi_sig</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-methods.html">GeneSet</a></span><span class="op">(</span><span class="va">epi_genes</span>, setName <span class="op">=</span> <span class="st">'THIERY_EPITHELIAL_CELLLINE'</span>, geneIdType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">epi_sig</span></code></pre></div>
<pre><code>## setName: THIERY_EPITHELIAL_CELLLINE 
## geneIds: ABCC3, ADAP1, ..., ZNF165 (total: 170)
## geneIdType: Symbol
## collectionType: Null 
## details: use 'details(object)'</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#检索间充质基因</span>
<span class="va">mes_genes</span> <span class="op">=</span> <span class="va">thiery_data</span><span class="op">$</span><span class="va">genes</span><span class="op">[</span><span class="va">thiery_data</span><span class="op">$</span><span class="va">epiMes_cellLine</span> <span class="op">%in%</span> <span class="st">'mes'</span><span class="op">]</span>
<span class="co">#过滤重复基因</span>
<span class="va">mes_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">mes_genes</span><span class="op">)</span>
<span class="co">#创建GeneSet对象</span>
<span class="va">mes_sig</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-methods.html">GeneSet</a></span><span class="op">(</span><span class="va">mes_genes</span>, setName <span class="op">=</span> <span class="st">'THIERY_MESENCHYMAL_CELLLINE'</span>, geneIdType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">mes_sig</span></code></pre></div>
<pre><code>## setName: THIERY_MESENCHYMAL_CELLLINE 
## geneIds: AKAP12, ANK2, ..., ZEB1 (total: 48)
## geneIdType: Symbol
## collectionType: Null 
## details: use 'details(object)'</code></pre>
<p>若感兴趣的基因集在分子标志数据库（MSigDB）中可用，则可以使用<code>msigdb</code>包获取。该包使用<code>ExperimentHub</code>将MSigDB中的基因集作为GeneSet对象嵌入<code><a href="https://rdrr.io/pkg/GSEABase/man/GeneSetCollection-methods.html">GSEABase::GeneSetCollection</a></code>对象中，而GeneSetCollection对象可以被用于存储多个GeneSet对象。<code>msigdb</code>包使用symbols或Entrez IDs为人和鼠提供标志基因（使用WEHI Smyth实验室提供的版本）。该包提供了7.2及以上版本的MSigDB。</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/msigdb">msigdb</a></span><span class="op">)</span>

<span class="co">#加载MSigDB基因集</span>
<span class="va">msigdb.hs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/getMsigdb.html">getMsigdb</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">'hs'</span>, id <span class="op">=</span> <span class="st">'SYM'</span>, version <span class="op">=</span> <span class="st">'7.2'</span><span class="op">)</span>
<span class="co">#添加KEGG基因集</span>
<span class="va">msigdb.hs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/appendKEGG.html">appendKEGG</a></span><span class="op">(</span><span class="va">msigdb.hs</span><span class="op">)</span>
<span class="va">msigdb.hs</span></code></pre></div>
<pre><code>## GeneSetCollection
##   names: chr11q, chr6q, ..., KEGG_VIRAL_MYOCARDITIS (31508 total)
##   unique identifiers: AP001767.2, SLC22A9, ..., OR8U3 (40049 total)
##   types in collection:
##     geneIdType: SymbolIdentifier (1 total)
##     collectionType: BroadCollection (1 total)</code></pre>
</div>
<div id="singscore" class="section level2">
<h2 class="hasAnchor">
<a href="#singscore" class="anchor"></a>Singscore</h2>
<p>准备好数据和基因集后，我们现在可以使用singscore<span class="citation">(Foroutan et al. 2018)</span>针对给定的基因集对样本进行评分，来探究其分子表型。Singscore是一种基于秩次的单样本评分方法，由<code>singscore</code> R/Bioconductor包实现。评分的核心计算可以被概括为两步：计算样本内的基因排名，使用排名计算基因集得分。传入singscore的数据应经过长度偏差校正（如TPM,RPKM,RSEM）并应该过滤掉大多数样本中低表达的基因。此处使用的数据已经经过了这样的处理，因此无需进一步预处理。更多有关singscore长度偏差校正和过滤的信息可以在<span class="citation">(Bhuva et al. 2019)</span>中被找到。</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/singscore">singscore</a></span><span class="op">)</span>

<span class="co">#根据基因表达对基因排序</span>
<span class="va">eranks</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/rankGenes.html">rankGenes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">emt_se</span>, <span class="st">'logRPKM'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#计算上皮得分</span>
<span class="va">epi_score</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">eranks</span>, <span class="va">epi_sig</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">epi_score</span><span class="op">)</span></code></pre></div>
<pre><code>##                  TotalScore TotalDispersion
## MDA468_Ctrl_Rep1  0.3230845        3992.642
## MDA468_Ctrl_Rep2  0.3222157        4030.448
## MDA468_Ctrl_Rep3  0.3214232        4120.145
## MDA468_EGF_Rep1   0.3230362        4076.409
## MDA468_EGF_Rep2   0.3218094        4169.812
## MDA468_EGF_Rep3   0.3274617        3951.870</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#计算间充质得分</span>
<span class="va">mes_score</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">eranks</span>, <span class="va">mes_sig</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mes_score</span><span class="op">)</span></code></pre></div>
<pre><code>##                    TotalScore TotalDispersion
## MDA468_Ctrl_Rep1 -0.027817504       10014.963
## MDA468_Ctrl_Rep2 -0.032615987       10333.722
## MDA468_Ctrl_Rep3 -0.030163516       10885.249
## MDA468_EGF_Rep1   0.014965710       10068.337
## MDA468_EGF_Rep2   0.008487366       10682.133
## MDA468_EGF_Rep3   0.018875730        8914.874</code></pre>
<p>不同于其他单样本方法，singscore的结果包含两种值：分数和离散度。我们发现这两种值都有助于解释基因集得分。得分表示了基因集中所有基因相对于所有测量基因的中位值表达，而离散度是衡量基因集中的基因表达协调性的指标，低离散度表示基因集中的基因有相似的表达水平。可以使用函数<code><a href="https://davislaboratory.github.io/singscore/reference/plotDispersion.html">plotDispersion()</a></code>绘制得分vs离散度的散点图，离散或连续型变量都可以被用于样本注释。<code>singscore</code>包中的所有绘图函数都可以通过设置<code>isInteractive = TRUE</code>来绘制交互图像。</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/plotDispersion.html">plotDispersion</a></span><span class="op">(</span>
  <span class="va">epi_score</span>,
  annot <span class="op">=</span> <span class="va">emt_se</span><span class="op">$</span><span class="va">Subline</span>,
  annot_name <span class="op">=</span> <span class="st">'Subline'</span>,
  size <span class="op">=</span> <span class="fl">5</span>,
  alpha <span class="op">=</span> <span class="fl">0.5</span>,
  isInteractive <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>从上图可以看出，MDA-MB-468细胞系更高的得分显示其相比于其他细胞系，具有更显著的上皮样特征。而这些样本中较低的离散度表明相对于其他基因，这些上皮基因有相似的表达水平。间充质基因集则显示出相反的模式，PMC42-ET和PMC42-LA细胞系相对于MDA-MB-468细胞系有更高的得分和更低的离散度。</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/plotDispersion.html">plotDispersion</a></span><span class="op">(</span>
  <span class="va">mes_score</span>,
  annot <span class="op">=</span> <span class="va">emt_se</span><span class="op">$</span><span class="va">Subline</span>,
  annot_name <span class="op">=</span> <span class="st">'Subline'</span>,
  size <span class="op">=</span> <span class="fl">5</span>,
  alpha <span class="op">=</span> <span class="fl">0.5</span>,
  isInteractive <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>研究基因集之间的关联性通常很有用，尤其是当表型之间预期存在生物学相关性时。<span class="citation">(Cursons et al. 2018)</span>和<span class="citation">(Foroutan et al. 2018)</span>利用上皮-间充质全景图更好地理解了生物模型在上皮-间充质转化上的行为。此外，在生物学中经常将生物模型与病人数据比较，从而对发现进行更好地翻译。Singscore通过在背景数据集中绘制细胞系数据来实现这一点。下图中，我们在背景中绘制了来自TCGA的乳腺癌病人数据，随后我们将我们的细胞系数据投射到病人数据上，以探究并解释上皮间充质全景的各个方面。通过将细胞系数据投射到病人数据上，细胞系上的发现可以被传递到病人层面。我们在此使用了预先用<code>singscore</code>包计算得到的TCGA乳腺癌样本数据的上皮和间充质得分。</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#加载预先计算好的TCGA乳腺癌EMT得分</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"scoredf_tcga_epi"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"scoredf_tcga_mes"</span><span class="op">)</span>

<span class="co">#绘制EMT全景图</span>
<span class="va">p_tcga</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/plotScoreLandscape.html">plotScoreLandscape</a></span><span class="op">(</span>
  scoredf1 <span class="op">=</span> <span class="va">scoredf_tcga_epi</span>,
  scoredf2 <span class="op">=</span> <span class="va">scoredf_tcga_mes</span>,
  scorenames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Epithelial score'</span>, <span class="st">'Mesenchymal score'</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p>下图显示了PMC42-ET和PMC42-LA有相似的间充质得分，但上皮得分截然不同。另一方面，MDA-MB-468则是非常偏向上皮样的细胞系，与间充质样细胞相去甚远。我们合理怀疑PMC42-LA兼具上皮和间充质两种特性。与singscore包中的其他绘图函数一样，下图也可以通过设置<code>isInteractive = TRUE</code>来添加交互功能。</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/projectScoreLandscape.html">projectScoreLandscape</a></span><span class="op">(</span>
  <span class="va">p_tcga</span>,
  scoredf1 <span class="op">=</span> <span class="va">epi_score</span>,
  scoredf2 <span class="op">=</span> <span class="va">mes_score</span>,
  annot <span class="op">=</span> <span class="va">emt_se</span><span class="op">$</span><span class="va">Subline</span>,
  annot_name <span class="op">=</span> <span class="st">'Subline'</span>,
  isInteractive <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>使用处理信息对样本进行注释，结果显示对MDA-MB-468进行缺氧处理会增加细胞系的间充质特性。</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/projectScoreLandscape.html">projectScoreLandscape</a></span><span class="op">(</span>
  <span class="va">p_tcga</span>,
  scoredf1 <span class="op">=</span> <span class="va">epi_score</span>,
  scoredf2 <span class="op">=</span> <span class="va">mes_score</span>,
  annot <span class="op">=</span> <span class="va">emt_se</span><span class="op">$</span><span class="va">Treatment</span>,
  annot_name <span class="op">=</span> <span class="st">'Treatment'</span>,
  isInteractive <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p>MSigDB的hallmark集合代表了与癌症进展相关的关键分子进程。其中HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION代表了癌症中的EMT，因此我们可以用它来计算样本的得分。我们可以用hallmark EMT得分注释样本，从而评估其与EMT景观的关系。下图显示hallmark EMT基因集可以表征上皮-间充质联合轴。</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#提取hallmark EMT标志基因</span>
<span class="va">hemt_sig</span> <span class="op">=</span> <span class="va">msigdb.hs</span><span class="op">[[</span><span class="st">'HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION'</span><span class="op">]</span><span class="op">]</span>
<span class="co">#使用标志基因对样本打分</span>
<span class="va">hemt_score</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">eranks</span>, <span class="va">hemt_sig</span><span class="op">)</span>

<span class="co">#绘制hallmark EMT基因集得分</span>
<span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/projectScoreLandscape.html">projectScoreLandscape</a></span><span class="op">(</span>
  <span class="va">p_tcga</span>,
  scoredf1 <span class="op">=</span> <span class="va">epi_score</span>,
  scoredf2 <span class="op">=</span> <span class="va">mes_score</span>,
  annot <span class="op">=</span> <span class="va">hemt_score</span><span class="op">$</span><span class="va">TotalScore</span>,
  annot_name <span class="op">=</span> <span class="st">'Hallmark EMT score'</span>,
  isInteractive <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<p>上图表明PMC42-ET和PMC42-LA之间的差异主要在上皮轴方向。我们可以使用函数<code><a href="https://davislaboratory.github.io/singscore/reference/plotRankDensity.html">plotRankDensity()</a></code>来研究单个样本内每个基因的贡献度。归一化的基因秩分布表明上皮基因集大多数的基因相比于其他基因，在PMC42-LA中有更高的表达水平。相比之下，上皮基因在PMC42-ET中的表达更为分散。这些图的交互式版本可以被用于探究单个基因。</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/plotRankDensity.html">plotRankDensity</a></span><span class="op">(</span><span class="va">eranks</span><span class="op">[</span>, <span class="st">'PMC42LA_Ctrl_Rep1'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, <span class="va">epi_sig</span>, isInteractive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/plotRankDensity.html">plotRankDensity</a></span><span class="op">(</span><span class="va">eranks</span><span class="op">[</span>, <span class="st">'PMC42ET_Ctrl_Rep1'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, <span class="va">epi_sig</span>, isInteractive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-17-2.png" width="700"></p>
</div>
<div id="stingscore" class="section level2">
<h2 class="hasAnchor">
<a href="#stingscore" class="anchor"></a>Stingscore</h2>
<p>有时由于成本或高质量基因组材料的可用性（如FFPE样本的情况），全转录组测量并不可行。这种情况下，我们希望能使用具有靶向性的转录组，然而大多数基因集评分方法（包括singscore）要求全转录组测序，因此不能在靶向测序背景中使用。此时，<code>singscore</code>包中的stingscore<span class="citation">(Bhuva, Cursons, and Davis 2020)</span>方法可以发挥作用。Stingscore使用一组稳定表达的基因在减少的测序环境中预估基因集得分。</p>
<p>为了演示stingscore在靶向测序环境中的使用，我们将使用上皮基因，间充质基因和一些稳定表达的基因来模拟一个靶向测序平台。我们将使用在<span class="citation">(Bhuva, Cursons, and Davis 2020)</span>中鉴定的top 5稳定表达的基因来对样本打分。</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#创建靶向测序基因列表</span>
<span class="va">sample_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/union.html">union</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html">geneIds</a></span><span class="op">(</span><span class="va">epi_sig</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html">geneIds</a></span><span class="op">(</span><span class="va">mes_sig</span><span class="op">)</span><span class="op">)</span>
<span class="va">sample_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/union.html">union</a></span><span class="op">(</span><span class="va">sample_genes</span>, <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/getStableGenes.html">getStableGenes</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span>
<span class="co">#选取测序数据中存在的靶向测序基因</span>
<span class="va">sample_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="va">sample_genes</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">emt_se</span><span class="op">)</span><span class="op">)</span>
<span class="co">#模拟靶向测序数据</span>
<span class="va">targeted_se</span> <span class="op">=</span> <span class="va">emt_se</span><span class="op">[</span><span class="va">sample_genes</span>, <span class="op">]</span>
<span class="va">targeted_se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 214 21 
## metadata(0):
## assays(2): counts logRPKM
## rownames(214): ABCC3 ADAP1 ... TARDBP HNRNPK
## rowData names(7): Chr Start ... gene_name gene_biotype
## colnames(21): MDA468_Ctrl_Rep1 MDA468_Ctrl_Rep2 ... PMC42LA_EGF_Rep2
##   PMC42LA_EGF_Rep3
## colData names(14): group lib.size ... Organism SRA.Study</code></pre>
<p>我们从原始数据中对感兴趣的基因取样，最终得到了包含214个基因的测序数据。事实上，这种大小的测序结果可以通过NanoString’s nCounter等平台测量得到。此处使用的前5个稳定表达的基因可以被用来对样本进行评分，并对表达数据标准化以进行其他分析。Singscore和stingscore唯一的区别就在于二者计算秩的方式。在stingscore中，基因的秩是通过稳定表达基因来预估的，预估后，与singscore相似，即可使用函数<code><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore()</a></code>计算最终得分。</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">st_genes</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/getStableGenes.html">getStableGenes</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span>
<span class="va">st_genes</span></code></pre></div>
<pre><code>## [1] "RBM45"  "BRAP"   "CIAO1"  "TARDBP" "HNRNPK"</code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#根据稳定表达基因预估基因秩</span>
<span class="va">st_eranks</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/rankGenes.html">rankGenes</a></span><span class="op">(</span><span class="va">targeted_se</span>, stableGenes <span class="op">=</span> <span class="va">st_genes</span><span class="op">)</span>

<span class="co">#使用模拟靶向数据对样本打分</span>
<span class="va">epi_score_st</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">st_eranks</span>, <span class="va">epi_sig</span><span class="op">)</span>
<span class="va">mes_score_st</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore</a></span><span class="op">(</span><span class="va">st_eranks</span>, <span class="va">mes_sig</span><span class="op">)</span>

<span class="co">#绘制singscore vs stingscore的散点图</span>
<span class="va">colmap</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'MDA-MB-468'</span> <span class="op">=</span> <span class="st">'#1B9E77'</span>, <span class="st">'PMC42-ET'</span> <span class="op">=</span> <span class="st">'#D95F02'</span>, <span class="st">'PMC42-LA'</span> <span class="op">=</span> <span class="st">'#7570B3'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span>
  <span class="va">epi_score</span><span class="op">$</span><span class="va">TotalScore</span>,
  <span class="va">epi_score_st</span><span class="op">$</span><span class="va">TotalScore</span>,
  col <span class="op">=</span> <span class="va">colmap</span><span class="op">[</span><span class="va">targeted_se</span><span class="op">$</span><span class="va">Subline</span><span class="op">]</span>,
  main <span class="op">=</span> <span class="st">'Epithelial score'</span>,
  xlab <span class="op">=</span> <span class="st">'Singscore'</span>,
  ylab <span class="op">=</span> <span class="st">'Stingscore (targeted)'</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>coef <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, col <span class="op">=</span> <span class="fl">2</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span>
  <span class="va">mes_score</span><span class="op">$</span><span class="va">TotalScore</span>,
  <span class="va">mes_score_st</span><span class="op">$</span><span class="va">TotalScore</span>,
  col <span class="op">=</span> <span class="va">colmap</span><span class="op">[</span><span class="va">targeted_se</span><span class="op">$</span><span class="va">Subline</span><span class="op">]</span>,
  main <span class="op">=</span> <span class="st">'Mesenchymal score'</span>,
  xlab <span class="op">=</span> <span class="st">'Singscore'</span>,
  ylab <span class="op">=</span> <span class="st">'Stingscore (targeted)'</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>coef <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, col <span class="op">=</span> <span class="fl">2</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<p>上图显示singscore和stingscore计算的上皮，间充质得分均高度线性相关。然而尽管得分高度线性相关，由于存在偏移，二者的值并不完全匹配。但因为样本的相对顺序不变，生物学推断不受由stingscore计算得分引起的偏移的影响。</p>
</div>
<div id="multiscore" class="section level2">
<h2 class="hasAnchor">
<a href="#multiscore" class="anchor"></a>Multiscore</h2>
<p>当假设未被预先确定时，我们可能希望使用多种基因集进行探索性分析。在这种情况下，我们会希望能针对100s-1000s个基因集对样本进行评分，并使用下游分析进行总结概况，此时可以使用函数<code><a href="https://davislaboratory.github.io/singscore/reference/multiScore.html">multiScore()</a></code>针对多个基因集同时对样本进行评分。这种实现方法比<code><a href="https://davislaboratory.github.io/singscore/reference/simpleScore.html">simpleScore()</a></code>快得多，因此是使用多基因集评分的推荐函数。</p>
<p>我们将针对来自hallmark集合，GO，KEGG通路，Reactome通路以及Biocarta通路的基因集对样本进行评分。可以使用<code>msigdb</code>包中的函数<code><a href="https://rdrr.io/pkg/msigdb/man/subsetCollection.html">subsetCollection()</a></code>来根据MSigDB类别及子类别选取GeneSetCollection的子集。该函数将返回GeneSetCollection对象，这些对象可以作为正常列表被串联在一起，但在将最终结果传递给函数<code><a href="https://davislaboratory.github.io/singscore/reference/multiScore.html">multiScore()</a></code>前，需要确保列表已经被转化回GeneSetCollection对象。</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#创建感兴趣的基因集集合</span>
<span class="va">genesigs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
  <span class="va">epi_sig</span>,
  <span class="va">mes_sig</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/subsetCollection.html">subsetCollection</a></span><span class="op">(</span><span class="va">msigdb.hs</span>, <span class="st">'h'</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/subsetCollection.html">subsetCollection</a></span><span class="op">(</span><span class="va">msigdb.hs</span>, <span class="st">'c2'</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'CP:KEGG'</span>, <span class="st">'CP:REACTOME'</span>, <span class="st">'CP:BIOCARTA'</span><span class="op">)</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/subsetCollection.html">subsetCollection</a></span><span class="op">(</span><span class="va">msigdb.hs</span>, <span class="st">'c5'</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'GO:BP'</span>, <span class="st">'GO:MF'</span>, <span class="st">'GO:CC'</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">genesigs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSetCollection-methods.html">GeneSetCollection</a></span><span class="op">(</span><span class="va">genesigs</span><span class="op">)</span></code></pre></div>
<p>函数<code><a href="https://davislaboratory.github.io/singscore/reference/multiScore.html">multiScore()</a></code>的返回结果是一个包含了得分矩阵和离散度矩阵的列表。由于某些基因集可能完全由数据中未测量的基因组成，这些矩阵中有可能会存在缺失值。</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">eranks</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/rankGenes.html">rankGenes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">emt_se</span>, <span class="st">'logRPKM'</span><span class="op">)</span><span class="op">)</span>
<span class="va">msigdb_scores</span> <span class="op">=</span> <span class="fu"><a href="https://davislaboratory.github.io/singscore/reference/multiScore.html">multiScore</a></span><span class="op">(</span><span class="va">eranks</span>, <span class="va">genesigs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">msigdb_scores</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## $Scores
##                                  MDA468_Ctrl_Rep1 MDA468_Ctrl_Rep2
## THIERY_EPITHELIAL_CELLLINE              0.3230845       0.32221565
## THIERY_MESENCHYMAL_CELLLINE            -0.0278175      -0.03261599
## HALLMARK_TNFA_SIGNALING_VIA_NFKB        0.1893111       0.18454327
## HALLMARK_HYPOXIA                        0.1549132       0.15766930
## HALLMARK_CHOLESTEROL_HOMEOSTASIS        0.3035149       0.30298773
## 
## $Dispersions
##                                  MDA468_Ctrl_Rep1 MDA468_Ctrl_Rep2
## THIERY_EPITHELIAL_CELLLINE               3992.642         4030.448
## THIERY_MESENCHYMAL_CELLLINE             10014.963        10333.722
## HALLMARK_TNFA_SIGNALING_VIA_NFKB         7278.825         7191.351
## HALLMARK_HYPOXIA                         7614.634         7633.907
## HALLMARK_CHOLESTEROL_HOMEOSTASIS         4222.445         4188.345</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">msigdb_scores</span>, <span class="va">dim</span><span class="op">)</span></code></pre></div>
<pre><code>## $Scores
## [1] 21043    21
## 
## $Dispersions
## [1] 21043    21</code></pre>
<p>通过绘制所有基因集的得分vs离散度可以展示这两种值之间的一般趋势。这种趋势是预料之中的，因为只有当基因集中的基因都协调表达时才能获得强正或强负的得分。更有趣的是那些有着低离散度和评分接近0的基因集，这些基因集包含的协调表达基因相对于所有被测量基因平均表达。</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/smoothScatter.html">smoothScatter</a></span><span class="op">(</span><span class="va">msigdb_scores</span><span class="op">$</span><span class="va">Scores</span>, <span class="va">msigdb_scores</span><span class="op">$</span><span class="va">Dispersions</span>, xlab <span class="op">=</span> <span class="st">'Score'</span>, ylab <span class="op">=</span> <span class="st">'Dispersion'</span>, main <span class="op">=</span> <span class="st">'Score vs Dispersion'</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-22-1.png" width="700"></p>
<p>我们可以通过评估单个样本的基因集来确定它们的表型组成。由于转录组测量在相对评估时更具有生物学意义，我们计算了一个PMC42-ET对照重复和另一个PMC42-LA对照重复的得分差异。下面列出了最高的正分和负分。</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#选择一个缺氧处理的样本和得分</span>
<span class="va">et_la_scores</span> <span class="op">=</span> <span class="va">msigdb_scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">[</span>, <span class="st">'PMC42ET_Ctrl_Rep1'</span><span class="op">]</span> <span class="op">-</span> <span class="va">msigdb_scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">[</span>, <span class="st">'PMC42LA_Ctrl_Rep1'</span><span class="op">]</span>
<span class="va">et_la_scores</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">et_la_scores</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">et_la_scores</span><span class="op">)</span></code></pre></div>
<pre><code>##                                                 GO_REGULATION_OF_EPINEPHRINE_SECRETION 
##                                                                             -0.3560168 
##                                           GO_POSITIVE_REGULATION_OF_MESODERM_FORMATION 
##                                                                             -0.3180387 
##                      GO_REGULATION_OF_VERY_LOW_DENSITY_LIPOPROTEIN_PARTICLE_REMODELING 
##                                                                             -0.3061849 
## GO_GAP_JUNCTION_CHANNEL_ACTIVITY_INVOLVED_IN_CELL_COMMUNICATION_BY_ELECTRICAL_COUPLING 
##                                                                             -0.3041113 
##                                           GO_INTERMEDIATE_DENSITY_LIPOPROTEIN_PARTICLE 
##                                                                             -0.2954161 
##                                                 HP_DECREASED_SERUM_COMPLEMENT_FACTOR_I 
##                                                                             -0.2910616</code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="va">et_la_scores</span><span class="op">)</span></code></pre></div>
<pre><code>##                            BIOCARTA_CYTOKINE_PATHWAY 
##                                            0.2212715 
## GO_POSITIVE_REGULATION_OF_T_CELL_TOLERANCE_INDUCTION 
##                                            0.2223007 
##  GO_CORTICOTROPIN_RELEASING_HORMONE_RECEPTOR_BINDING 
##                                            0.2240487 
##          GO_REGULATION_OF_LIPID_TRANSPORTER_ACTIVITY 
##                                            0.2332605 
##                            GO_11_CIS_RETINAL_BINDING 
##                                            0.2378071 
##                   GO_ESTROGEN_2_HYDROXYLASE_ACTIVITY 
##                                            0.2410252</code></pre>
</div>
</div>
<div id="识别并可视化高阶表型" class="section level1">
<h1 class="hasAnchor">
<a href="#%E8%AF%86%E5%88%AB%E5%B9%B6%E5%8F%AF%E8%A7%86%E5%8C%96%E9%AB%98%E9%98%B6%E8%A1%A8%E5%9E%8B" class="anchor"></a>识别并可视化高阶表型</h1>
<p>在很多RNA-seq测序数据分析中，基因集富集分析（可以是一个功能性类别评分手段的过表达分析）通常会被使用并产生一系列基因集，附有相关统计量，p值或假阳性率（FDR）。在大多数传统合作中，这些结果通常由领域专家或生物学家手动校正或扫描，接着他们会提取与其特定实验或感兴趣领域相关的主题。考虑到大多数基因集富集分析会产生100s到1000s 个基因集，手动管理和解释可能非常耗时。</p>
<p>本分析流程中使用的方法R包，<code>vissE</code>，可以根据数据自动提取生物学主题，从而减少领域专家的负担。以下分析使用@cursons15 数据集的子集（确切的说是来自PMC42-ET和PMC42-LA亚系的未处理样本）来演示vissE的分析流程。我们首先进行差异表达（DE）分析以评估PMC42-LA和PMC42-ET亚系之间的差异。然后使用流行的<code><a href="https://rdrr.io/pkg/limma/man/roast.html">limma::fry</a></code>方法进行功能性分析，从而识别跨亚系不同的生物过程。最后，我们使用vissE鉴定、总结和可视化高阶分子表型。</p>
<div id="差异表达分析" class="section level2">
<h2 class="hasAnchor">
<a href="#%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE%E5%88%86%E6%9E%90" class="anchor"></a>差异表达分析</h2>
<p>由于我们对了解PMC42亚系之间的差异的功能意义感兴趣，我们将仅使用<span class="citation">(Cursons et al. 2015)</span>中来自PMC42亚系的对照样本。在处理计数数据时，我们将使用来自<span class="citation">(Chen, Lun, and Smyth 2016)</span>的拟似然差异表达分析流程。我们推荐使用者参考分析流程的原始出版物，以了解在分析的每个步骤中做出的决策的细节。</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma">limma</a></span><span class="op">)</span>

<span class="co">#从原始数据中提取PMC42-ET和PMC42-LA的对照样本</span>
<span class="va">emt_se_sub</span> <span class="op">=</span> <span class="va">emt_se</span><span class="op">[</span>, <span class="va">emt_se</span><span class="op">$</span><span class="va">Subline</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'PMC42-ET'</span>, <span class="st">'PMC42-LA'</span><span class="op">)</span> <span class="op">&amp;</span>
                      <span class="va">emt_se</span><span class="op">$</span><span class="va">Treatment</span> <span class="op">==</span> <span class="st">'control'</span><span class="op">]</span>
<span class="co">#从SE对象中创建DGEList对象</span>
<span class="va">emt_dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/emtdata/man/asDGEList.html">asDGEList</a></span><span class="op">(</span><span class="va">emt_se_sub</span><span class="op">)</span>
<span class="co">#创建模型设计矩阵</span>
<span class="va">design</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Subline</span>, data <span class="op">=</span> <span class="va">emt_dge</span><span class="op">$</span><span class="va">samples</span><span class="op">)</span>
<span class="co">#去除低表达基因</span>
<span class="va">keep</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html">filterByExpr</a></span><span class="op">(</span><span class="va">emt_dge</span>, design <span class="op">=</span> <span class="va">design</span><span class="op">)</span>
<span class="va">emt_dge</span> <span class="op">=</span> <span class="va">emt_dge</span><span class="op">[</span><span class="va">keep</span>, <span class="op">]</span>
<span class="co">#使用TMM计算标准化因子</span>
<span class="va">emt_dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span><span class="op">(</span><span class="va">emt_dge</span>, method <span class="op">=</span> <span class="st">"TMM"</span><span class="op">)</span>

<span class="co">#估计离散度</span>
<span class="va">emt_dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateDisp.html">estimateDisp</a></span><span class="op">(</span><span class="va">emt_dge</span>, design <span class="op">=</span> <span class="va">design</span>, robust <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#使用edgeR中带有拟似然检验（GLM）函数的广义线性模型进行拟合</span>
<span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html">glmQLFit</a></span><span class="op">(</span><span class="va">emt_dge</span>, <span class="va">design</span>, robust <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#获取差异表达（DE）基因列表</span>
<span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html">glmQLFTest</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="st">'SublinePMC42-LA'</span><span class="op">)</span>
<span class="va">is.de</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/decidetestsDGE.html">decideTestsDGE</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMD.html">plotMD</a></span><span class="op">(</span><span class="va">fit</span>, status <span class="op">=</span> <span class="va">is.de</span>, main <span class="op">=</span> <span class="st">'PMC42-LA vs. PMC42-ET'</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#以p值0.05为阈值筛选DE基因列表</span>
<span class="va">DE_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html">topTags</a></span><span class="op">(</span><span class="va">res</span>, n <span class="op">=</span> <span class="cn">Inf</span>, p.value <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">DE_list</span><span class="op">)</span></code></pre></div>
<pre><code>##          gene_biotype     logFC   logCPM         F       PValue          FDR
## FAT2   protein_coding -7.588511 6.614230 1334.7019 5.576353e-13 1.346410e-08
## TGFBR2 protein_coding -9.993858 3.596306 1040.0713 2.218916e-12 1.759046e-08
## SPINT1 protein_coding  6.890147 5.922593  985.9245 2.982160e-12 1.759046e-08
## PLS3   protein_coding -6.059286 5.070152  958.2729 3.489918e-12 1.759046e-08
## PLTP   protein_coding -6.202561 5.935153  950.8729 3.642671e-12 1.759046e-08
## KLK11  protein_coding  8.779300 3.708962  877.8718 5.662833e-12 2.278818e-08</code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#获取所有基因的统计量</span>
<span class="va">allg_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html">topTags</a></span><span class="op">(</span><span class="va">res</span>, n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
</div>
<div id="使用limmafry测试基因集" class="section level2">
<h2 class="hasAnchor">
<a href="#%E4%BD%BF%E7%94%A8limmafry%E6%B5%8B%E8%AF%95%E5%9F%BA%E5%9B%A0%E9%9B%86" class="anchor"></a>使用<code>limma::fry</code>测试基因集</h2>
<p>使用<code><a href="https://rdrr.io/pkg/limma/man/roast.html">limma::fry</a></code>进行功能分析，以鉴定跨PMC42亚系发生改变的生物过程。用于该分析的基因集来自分子特征数据库（MSigDB）<span class="citation">Liberzon et al. (2015)</span>，并可通过<code>msigdb</code>包访问，如上一节所示。</p>
<p>根据我们使用vissE的一般经验，我们发现精选的基因集集合，例如hallmark集合（h）、curated通路集合（c2）和GO集合（GO），在使用时最能有助于深入了解生物学过程。此外，我们选取了感兴趣的子集的集合，然而感兴趣的集合/子集合的选择将取决于正在研究的生物学方向。例如，在研究免疫过程时可以选择免疫特征集合（c7）。</p>
<p>使用<code><a href="https://rdrr.io/pkg/limma/man/roast.html">limma::fry</a></code>进行基因集检测的一个常用方法是对每个集合分别进行分析，从而校正每个集合中的多假设检验。当生物学的理解是来自一个独立于其他集合的集合中时，这是一个有效的方法。然而，由于我们使用vissE依赖所有的集合对相关方向进行深入了解，我们合并基因集并对所有的基因集集合使用<code><a href="https://rdrr.io/pkg/limma/man/roast.html">limma::fry</a></code>。</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#选取感兴趣的基因集/子集</span>
<span class="va">genesigs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
  <span class="va">epi_sig</span>,
  <span class="va">mes_sig</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/subsetCollection.html">subsetCollection</a></span><span class="op">(</span><span class="va">msigdb.hs</span>, <span class="st">'h'</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'CP:KEGG'</span>, <span class="st">'CP:REACTOME'</span>, <span class="st">'CP:BIOCARTA'</span>, <span class="st">'GO:BP'</span>, <span class="st">'GO:MF'</span>, <span class="st">'GO:CC'</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">genesigs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSetCollection-methods.html">GeneSetCollection</a></span><span class="op">(</span><span class="va">genesigs</span><span class="op">)</span>

<span class="co">#使用此前生成的基因集集合（genesigs）创建fry index</span>
<span class="va">fry_indices</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ids2indices.html">ids2indices</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html">geneIds</a></span><span class="op">(</span><span class="va">genesigs</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">emt_dge</span><span class="op">)</span><span class="op">)</span>
<span class="co">#去除极小或极大的基因集</span>
<span class="va">fry_indices</span> <span class="op">=</span> <span class="va">fry_indices</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">fry_indices</span>, <span class="va">length</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">10</span>
                          <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">fry_indices</span>, <span class="va">length</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">500</span><span class="op">]</span>

<span class="co">#使用limma::fry进行基因集富集分析</span>
<span class="va">fry_res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/roast.html">fry</a></span><span class="op">(</span><span class="va">emt_dge</span>,
              index <span class="op">=</span> <span class="va">fry_indices</span>,
              design <span class="op">=</span> <span class="va">design</span>,
              contrast <span class="op">=</span> <span class="st">'SublinePMC42-LA'</span><span class="op">)</span>

<span class="co">#选择显著性结果（FDR &lt; 0.05）</span>
<span class="va">fry_res</span> <span class="op">=</span> <span class="va">fry_res</span><span class="op">[</span><span class="va">fry_res</span><span class="op">$</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">fry_res</span><span class="op">)</span></code></pre></div>
<pre><code>##                                                         NGenes Direction
## GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS                     13        Up
## GO_LABYRINTHINE_LAYER_MORPHOGENESIS                         18        Up
## GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION                16      Down
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION     15        Up
## GO_DNA_POLYMERASE_COMPLEX                                   21      Down
## REACTOME_RA_BIOSYNTHESIS_PATHWAY                            16        Up
##                                                               PValue
## GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS                 7.343476e-07
## GO_LABYRINTHINE_LAYER_MORPHOGENESIS                     2.604803e-06
## GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION            3.526290e-06
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION 3.787180e-06
## GO_DNA_POLYMERASE_COMPLEX                               4.208996e-06
## REACTOME_RA_BIOSYNTHESIS_PATHWAY                        4.238159e-06
##                                                                 FDR
## GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS                 0.004489003
## GO_LABYRINTHINE_LAYER_MORPHOGENESIS                     0.004489003
## GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION            0.004489003
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION 0.004489003
## GO_DNA_POLYMERASE_COMPLEX                               0.004489003
## REACTOME_RA_BIOSYNTHESIS_PATHWAY                        0.004489003
##                                                         PValue.Mixed
## GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS                 1.220636e-05
## GO_LABYRINTHINE_LAYER_MORPHOGENESIS                     3.708566e-07
## GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION            5.533273e-07
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION 3.109981e-06
## GO_DNA_POLYMERASE_COMPLEX                               6.086205e-06
## REACTOME_RA_BIOSYNTHESIS_PATHWAY                        3.192814e-06
##                                                            FDR.Mixed
## GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS                 1.981291e-05
## GO_LABYRINTHINE_LAYER_MORPHOGENESIS                     3.173336e-06
## GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION            3.608736e-06
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION 7.916704e-06
## GO_DNA_POLYMERASE_COMPLEX                               1.208747e-05
## REACTOME_RA_BIOSYNTHESIS_PATHWAY                        8.019966e-06</code></pre>
</div>
<div id="为visse准备基因集测试结果" class="section level2">
<h2 class="hasAnchor">
<a href="#%E4%B8%BAvisse%E5%87%86%E5%A4%87%E5%9F%BA%E5%9B%A0%E9%9B%86%E6%B5%8B%E8%AF%95%E7%BB%93%E6%9E%9C" class="anchor"></a>为vissE准备基因集测试结果</h2>
<p>利用差异表达分析和基因集富集分析产生的可用信息，我们现在应用vissE（Visualising Set Enrichment）分析来鉴定高阶分子表型。我们注意到，类似的分析可以使用通过各种其他基因集测试方法实现，例如来自<code>clusterProfiler</code>包或流行的GSEA工具的过表达分析（ORA）<span class="citation">(Subramanian et al. 2005)</span>。</p>
<p>一项标准的vissE分析包括3步：</p>
<ol style="list-style-type: decimal">
<li>计算基因集重叠和基因集重叠网络。</li>
<li>基于重叠网络识别基因集聚类。</li>
<li>使用文本挖掘分析表征聚类。</li>
<li>（可选）可视化每个基因集类群的基因水平统计数据。</li>
</ol>
<p>为了进行这些分析，我们首先要准备如下的必要数据:</p>
<ul>
<li>一个<code>GeneSetCollection</code>对象，通过从基因集富集分析确定的重要基因集中选取子集（本例中来自<code>limma:fry</code>）。</li>
<li>一个命名向量，包含用于可视化的基因集统计信息。</li>
<li>一个命名向量，包含DE分析中的基因水平统计信息。</li>
</ul>
<p>我们用于可视化目的的基因集统计数据是一个有正负的对数转换后的假阳性率(<span class="math inline">\(\pm 1\times -\log_{10}(FDR)\)</span>)，其中正代表上调基因集，负代表下调基因集。用于可视化的基因水平统计量是差异倍数的对数（logFC）。</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#通过对重要基因集进行子集化来创建 GeneSetCollection</span>
<span class="va">siggs</span> <span class="op">=</span> <span class="va">genesigs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">fry_res</span><span class="op">)</span><span class="op">]</span>
<span class="va">siggs</span></code></pre></div>
<pre><code>## GeneSetCollection
##   names: GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS, GO_LABYRINTHINE_LAYER_MORPHOGENESIS, ..., GO_PLATELET_ALPHA_GRANULE_LUMEN (1667 total)
##   unique identifiers: CS, SIRT4, ..., CDY2A (13781 total)
##   types in collection:
##     geneIdType: SymbolIdentifier (1 total)
##     collectionType: BroadCollection, NullCollection (2 total)</code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#准备用于可视化的基因集统计量（signed -log10(FDR)）</span>
<span class="va">gsStats</span> <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">fry_res</span><span class="op">$</span><span class="va">FDR</span><span class="op">)</span>
<span class="co">#使用正号表示上调的基因集</span>
<span class="va">gsStats</span> <span class="op">=</span> <span class="va">gsStats</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Up'</span> <span class="op">=</span> <span class="fl">1</span>, <span class="st">'Down'</span> <span class="op">=</span> <span class="op">-</span><span class="fl">1</span><span class="op">)</span><span class="op">[</span><span class="va">fry_res</span><span class="op">$</span><span class="va">Direction</span><span class="op">]</span>
<span class="co">#必须是一个命名向量</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gsStats</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">fry_res</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gsStats</span><span class="op">)</span></code></pre></div>
<pre><code>##                 GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS 
##                                                 2.34785 
##                     GO_LABYRINTHINE_LAYER_MORPHOGENESIS 
##                                                 2.34785 
##            GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION 
##                                                -2.34785 
## GO_POSITIVE_REGULATION_OF_T_HELPER_CELL_DIFFERENTIATION 
##                                                 2.34785 
##                               GO_DNA_POLYMERASE_COMPLEX 
##                                                -2.34785 
##                        REACTOME_RA_BIOSYNTHESIS_PATHWAY 
##                                                 2.34785</code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#为可视化准备基因水平统计数量(logFC)</span>
<span class="va">gStats</span> <span class="op">=</span> <span class="va">allg_list</span><span class="op">$</span><span class="va">logFC</span>
<span class="co">#必须是一个命名向量</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gStats</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">allg_list</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gStats</span><span class="op">)</span></code></pre></div>
<pre><code>##      FAT2    TGFBR2    SPINT1      PLS3      PLTP     KLK11 
## -7.588511 -9.993858  6.890147 -6.059286 -6.202561  8.779300</code></pre>
</div>
<div id="计算重叠网络" class="section level2">
<h2 class="hasAnchor">
<a href="#%E8%AE%A1%E7%AE%97%E9%87%8D%E5%8F%A0%E7%BD%91%E7%BB%9C" class="anchor"></a>计算重叠网络</h2>
<p>vissE包支持两种基于基因集之间基因重叠的基因集相似性度量：Jaccard指数（默认）和重叠系数。重叠系数强调了基因集之间的亲子关系，因此是分层结构数据库（例如GO或Reactome）的理想选择，但是，仅仅推荐在单个分层结构数据库而不是组合式数据库上运行vissE时才这样做。</p>
<p>由于我们对MSigDB的不同集合进行了基因集测试，因此使用Jaccard指数计算基因集相似性。随后，我们使用基因集相似性数据创建一个网络，并将其可视化。</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/vissE">vissE</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org">igraph</a></span><span class="op">)</span>

<span class="co">#计算重要基因组之间的重叠</span>
<span class="va">gs_ovlap</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/computeMsigOverlap.html">computeMsigOverlap</a></span><span class="op">(</span><span class="va">siggs</span>, thresh <span class="op">=</span> <span class="fl">0.25</span>, measure <span class="op">=</span> <span class="st">'jaccard'</span><span class="op">)</span>
<span class="co">#利用重叠创建一个网络</span>
<span class="va">gs_ovnet</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/computeMsigNetwork.html">computeMsigNetwork</a></span><span class="op">(</span><span class="va">gs_ovlap</span>, msigGsc <span class="op">=</span> <span class="va">siggs</span><span class="op">)</span>
<span class="co">#可视化网络</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">21</span><span class="op">)</span> <span class="co">#设置随机数以便于获得重复结果</span>
<span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigNetwork.html">plotMsigNetwork</a></span><span class="op">(</span><span class="va">gs_ovnet</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-27-1.png" width="768"></p>
<p>上面的重叠网络图是基于MSigDB类别注释的，然而，通过使用基因集统计量注释节点可以包含更多信息。这里我们使用上一节准备的基因集统计量。</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#绘制网络并覆盖基因集统计量</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">21</span><span class="op">)</span> <span class="co">#设置随机数以便于获得重复结果</span>
<span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigNetwork.html">plotMsigNetwork</a></span><span class="op">(</span><span class="va">gs_ovnet</span>, genesetStat <span class="op">=</span> <span class="va">gsStats</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-28-1.png" width="768"></p>
</div>
<div id="确定网络中的群" class="section level2">
<h2 class="hasAnchor">
<a href="#%E7%A1%AE%E5%AE%9A%E7%BD%91%E7%BB%9C%E4%B8%AD%E7%9A%84%E7%BE%A4" class="anchor"></a>确定网络中的群</h2>
<p>给定一个相关基因集的网络，我们预计相关基因集可能代表一个常见的高阶生物过程。因此，我们对基因集进行聚类以识别基因集类群。 用户可以选择他们喜欢的聚类方法，但是，为了使用重叠图中提供的信息，我们在此推荐使用图聚类方法。 由于我们有一个基因集网络，我们可以使用诸如来自<code>igraph</code>R 包的那些图聚类算法。确切地说，我们推荐使用 walktrap 算法，因为它同时适用于密集和稀疏图形，并且擅长识别大小类群<span class="citation">(Yang, Algesheimer, and Tessone 2016)</span>。</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#使用图聚类算法识别聚类</span>
<span class="va">grps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cluster_walktrap.html">cluster_walktrap</a></span><span class="op">(</span><span class="va">gs_ovnet</span><span class="op">)</span>
<span class="co">#提取聚类结果</span>
<span class="va">grps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/groups.html">groups</a></span><span class="op">(</span><span class="va">grps</span><span class="op">)</span>
<span class="co">#移除小类群</span>
<span class="va">grps</span> <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">grps</span>, <span class="va">length</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">5</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">grps</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 43</code></pre>
<p>由于我们可以识别大量基因集类群，我们需要提出一个优先级方案来适当地探索它们。vissE的第二步对大型聚类最适用，因为有更多信息可用于进行文本挖掘分析，因此类群大小可用作优先级方案。然而，大型基因集类群可能只是从数据库中具有良好代表性的过程中产生的。例如，许多数据库过多地描述了免疫相关过程，因为过去对这些过程进行了广泛研究，因此免疫过程可能会形成大型基因集类群。为了避免选择这样的基因集类群，我们还使用基因集统计量对基因集进行了优先级排序。因此，我们建议采用组合优先排序方法。在此分析流程中，我们优先考虑同时具有较大平均统计量(<span class="math inline">\(\mathrm{mean}(-\log_{10}(FDR))\)</span>)的大型基因集类群。<span class="citation">(Bhuva, Cursons, and Davis 2020)</span>使用的等级乘积方法用于在对基因集类群进行排序时，对这两个度量进行同等加权。</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#计算类群大小</span>
<span class="va">grp_size</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">grps</span>, <span class="va">length</span><span class="op">)</span>
<span class="co">#计算类群统计量-绝对统计量的中位数</span>
<span class="va">grp_stat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">grps</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">gsStats</span><span class="op">[</span><span class="va">x</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#结合排序乘积</span>
<span class="va">grp_pr</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rank.html">rank</a></span><span class="op">(</span><span class="va">grp_size</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/rank.html">rank</a></span><span class="op">(</span><span class="va">grp_stat</span><span class="op">)</span>
<span class="co">#使用排序乘积对组进行排序-最大化类群的大小和统计量</span>
<span class="va">grps</span> <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">grp_pr</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<p>显然，从获得的基因集类群的数量来看，很难一次将所有基因集类群可视化，因此，我们将重点分析前12个优先的基因集类群。</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#绘制前12个类群</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">21</span><span class="op">)</span> <span class="co">#设置随机数以便于获得重复结果</span>
<span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigNetwork.html">plotMsigNetwork</a></span><span class="op">(</span><span class="va">gs_ovnet</span>, markGroups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>, genesetStat <span class="op">=</span> <span class="va">gsStats</span><span class="op">)</span>
<span class="va">p1</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-31-1.png" width="768"></p>
</div>
<div id="描述基因集聚类特征" class="section level2">
<h2 class="hasAnchor">
<a href="#%E6%8F%8F%E8%BF%B0%E5%9F%BA%E5%9B%A0%E9%9B%86%E8%81%9A%E7%B1%BB%E7%89%B9%E5%BE%81" class="anchor"></a>描述基因集聚类特征</h2>
<p>在对基因集进行聚类之后，我们想对每个类群进行生物学解释。由于来自MSigDB的基因集有合适的命名（即使是那些没有人工校正的基因集）并带有简短的描述，我们可以利用这些信息来评估每个类群的共同生物学主题。</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#所有基因集都有名字</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">siggs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, <span class="va">setName</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "GO_TRICARBOXYLIC_ACID_METABOLIC_PROCESS"     
## [2] "GO_LABYRINTHINE_LAYER_MORPHOGENESIS"         
## [3] "GO_DNA_UNWINDING_INVOLVED_IN_DNA_REPLICATION"</code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#几乎所有的基因集都有简短的描述</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">siggs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, <span class="va">description</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "The chemical reactions and pathways involving dicarboxylic acids, any organic acid containing three carboxyl (COOH) groups or anions (COO-). [GOC:mah]"                                        
## [2] "The process in which the labyrinthine layer of the placenta is generated and organized. [GOC:dph]"                                                                                             
## [3] "The process in which interchain hydrogen bonds between two strands of DNA are broken or 'melted', generating unpaired template strands for DNA replication. [ISBN:071673706X, ISBN:0815316194]"</code></pre>
<p>VissE通过对基因集名或描述进行文本挖掘分析，自动对每个聚类进行注释。在每个类群内计算通过逆文档频率（IDF）加权的词频，并使用词云进行可视化。这些词云可以用来自动注释基因集。</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#计算并绘制文本挖掘的结果：使用基因集名</span>
<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigWordcloud.html">plotMsigWordcloud</a></span><span class="op">(</span>msigGsc <span class="op">=</span> <span class="va">siggs</span>, groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">'Name'</span><span class="op">)</span>
<span class="va">p2</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-33-1.png" width="864"></p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#计算并绘制文本挖掘结果：使用基因集简短描述</span>
<span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigWordcloud.html">plotMsigWordcloud</a></span><span class="op">(</span>msigGsc <span class="op">=</span> <span class="va">siggs</span>, groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">'Short'</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-33-2.png" width="864"></p>
</div>
<div id="可视化基因集聚类的基因水平统计量" class="section level2">
<h2 class="hasAnchor">
<a href="#%E5%8F%AF%E8%A7%86%E5%8C%96%E5%9F%BA%E5%9B%A0%E9%9B%86%E8%81%9A%E7%B1%BB%E7%9A%84%E5%9F%BA%E5%9B%A0%E6%B0%B4%E5%B9%B3%E7%BB%9F%E8%AE%A1%E9%87%8F" class="anchor"></a>可视化基因集聚类的基因水平统计量</h2>
<p>在用生物学主题注释基因集聚类之后，下一步是进一步了解对这些基因集类群的显著性做出贡献的个体。在这里，我们将每个已鉴定的基因集类群的基因水平统计数据（例如，来自DE分析）联系起来。除了这些统计量之外，评估基因集类群中的基因频率也可能有用，即基因所属的类群内基因集的数量。在函数<code><a href="https://DavisLaboratory.github.io/vissE/reference/plotGeneStats.html">plotGeneStats()</a></code>中，我们根据基因频率绘制基因水平统计量。具有高统计量和高频率的基因可以被认为是鉴定的生物学过程中的关键基因。为了减少过度绘制，对x轴使用了抖动（由于其离散性）。</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#绘制基因水平统计量（logFC）</span>
<span class="va">p3</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotGeneStats.html">plotGeneStats</a></span><span class="op">(</span><span class="va">gStats</span>,
                   msigGsc <span class="op">=</span> <span class="va">siggs</span>,
                   groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>,
                   statName <span class="op">=</span> <span class="st">'logFC'</span><span class="op">)</span>
<span class="co">#由于所有vissE的图都是ggplot2对象，因此可以对其进行修改</span>
<span class="va">p3</span> <span class="op">=</span> <span class="va">p3</span> <span class="op">+</span> <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, colour <span class="op">=</span> <span class="fl">2</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="va">p3</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-34-1.png" width="864"></p>
</div>
<div id="评估聚类中的蛋白质互作ppi" class="section level2">
<h2 class="hasAnchor">
<a href="#%E8%AF%84%E4%BC%B0%E8%81%9A%E7%B1%BB%E4%B8%AD%E7%9A%84%E8%9B%8B%E7%99%BD%E8%B4%A8%E4%BA%92%E4%BD%9Cppi" class="anchor"></a>评估聚类中的蛋白质互作（PPI）</h2>
<p>与基因集一样，蛋白质间相互作用（PPI）网络也可用于评估基因组合的功能作用。参与蛋白质间相互作用的相关的基因很可能也参与相关功能。为了评估这种独立的证据，vissE提供了一种模块，用来评估基因集类群中基因之间的蛋白质间相互作用。</p>
<p>在为每个类群生成网络之前，需要加载完整的人类PPI网络。<code>msigdb</code>包提供了来自国际分子交换联盟(IMEx)处理过的PPI网络。人类和小鼠的PPIs是可用的，可以使用<code><a href="https://rdrr.io/pkg/msigdb/man/getIMEX.html">getIMEX()</a></code>函数加载。为了扩大覆盖范围，PPIs可以通过同源转换从不同的生物体中借用。这一点可以通过设置<code>inferred=TRUE</code>来实现。</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#查找存储的IMEx版本</span>
<span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/getIMEXVersions.html">getIMEXVersions</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "2021-07-06"</code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#使用msigdb包从IMEx数据库加载人类的PPI数据</span>
<span class="va">ppi</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/msigdb/man/getIMEX.html">getIMEX</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">'hs'</span>, inferred <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ppi</span><span class="op">)</span></code></pre></div>
<pre><code>##   EntrezA EntrezB Taxid InteractorA InteractorB SymbolA SymbolB
## 1       1    1026  9606      P04217      P38936    A1BG  CDKN1A
## 2       1   10549  9606      P04217      Q13162    A1BG   PRDX4
## 3       1    2886  9606      P04217      Q14451    A1BG    GRB7
## 4       1     310  9606      P04217      P20073    A1BG   ANXA7
## 5       1     368  9606      P04217      O95255    A1BG   ABCC6
## 6       1    6606  9606      P04217      Q16637    A1BG    SMN1
##        InteractionType DetectionMethod Confidence Inferred
## 1 physical association      two hybrid  0.3678499    FALSE
## 2 physical association      two hybrid  0.3678499    FALSE
## 3 physical association      two hybrid  0.3678499    FALSE
## 4 physical association      two hybrid  0.3678499    FALSE
## 5 physical association      two hybrid  0.3678499    FALSE
## 6 physical association      two hybrid  0.3678499    FALSE</code></pre>
<p>使用基因水平统计量来注释网络中的每个基因，以提供整体视图。可以使用以下过滤条件对网络进行过滤：</p>
<ol style="list-style-type: decimal">
<li>
<code>threshFrequency</code> - 将选择在一个基因集类群中具有特定频率的节点，即该基因所属的基因集在类群中的比例。</li>
<li>
<code>threshConfidence</code> - 将对PPI的边界进行过滤，以选择具有高置信度统计量的相互作用。</li>
<li>
<code>threshStatistic</code> - 将选择感兴趣的统计量(例如，倍数差异对数值)大于阈值的基因。</li>
</ol>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p4</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigPPI.html">plotMsigPPI</a></span><span class="op">(</span>
  ppidf <span class="op">=</span> <span class="va">ppi</span>,
  msigGsc <span class="op">=</span> <span class="va">siggs</span>,
  groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>,
  geneStat <span class="op">=</span> <span class="va">gStats</span>,
  threshFrequency <span class="op">=</span> <span class="fl">0.25</span>, <span class="co">#每个类群中基因频率阈值</span>
  threshConfidence <span class="op">=</span> <span class="fl">0.2</span>, <span class="co">#PPI边界置信度阈值</span>
  threshStatistic <span class="op">=</span> <span class="fl">0.5</span>, <span class="co">#基因水平统计量阈值</span>
  threshUseAbsolute <span class="op">=</span> <span class="cn">TRUE</span> <span class="co">#对绝对值应用阈值</span>
<span class="op">)</span>
<span class="va">p4</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-36-1.png" width="700"></p>
</div>
<div id="由visse解读生物学意义" class="section level2">
<h2 class="hasAnchor">
<a href="#%E7%94%B1visse%E8%A7%A3%E8%AF%BB%E7%94%9F%E7%89%A9%E5%AD%A6%E6%84%8F%E4%B9%89" class="anchor"></a>由vissE解读生物学意义</h2>
<p>现在我们已经生成了三个图，我们可以将它们结合起来以提供数据的整体视图。这将提供一个由分析鉴定的生物学主题的综合视图，以及驱动这些基因集的潜在基因。</p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span>

<span class="co">#将图组合成统一的可视化效果</span>
<span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span> <span class="op">+</span> <span class="va">p3</span> <span class="op">+</span> <span class="va">p4</span> <span class="op">+</span> <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html">plot_layout</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-37-1.png" width="1440"></p>
<p>该分析的预期结果在包含9个基因集的类群”90”中确定。我们可以专门为类群”90”生成上面的图，同时在基因水平统计图中注释更多的基因。</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#研究特定的类群</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">21</span><span class="op">)</span> <span class="co">#设置随机数以便于获得重复结果</span>
<span class="co">#选择要绘图的类群</span>
<span class="va">g</span> <span class="op">=</span> <span class="st">'90'</span>

<span class="co">#提取感兴趣的类群</span>
<span class="va">grps</span><span class="op">[</span><span class="st">'90'</span><span class="op">]</span></code></pre></div>
<pre><code>## $`90`
## [1] "GO_KERATINIZATION"                           
## [2] "GO_KERATINOCYTE_DIFFERENTIATION"             
## [3] "GO_EPIDERMAL_CELL_DIFFERENTIATION"           
## [4] "GO_CORNIFICATION"                            
## [5] "REACTOME_FORMATION_OF_THE_CORNIFIED_ENVELOPE"
## [6] "REACTOME_KERATINIZATION"                     
## [7] "GO_KERATIN_FILAMENT"                         
## [8] "GO_EPIDERMIS_DEVELOPMENT"                    
## [9] "GO_SKIN_DEVELOPMENT"</code></pre>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#为指定类群计算并绘制vissE面板</span>
<span class="va">p1_g</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigNetwork.html">plotMsigNetwork</a></span><span class="op">(</span><span class="va">gs_ovnet</span>, markGroups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="va">g</span><span class="op">]</span>, genesetStat <span class="op">=</span> <span class="va">gsStats</span><span class="op">)</span>
<span class="va">p2_g</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigWordcloud.html">plotMsigWordcloud</a></span><span class="op">(</span>msigGsc <span class="op">=</span> <span class="va">siggs</span>, groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="va">g</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">'Short'</span><span class="op">)</span>
<span class="va">p3_g</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotGeneStats.html">plotGeneStats</a></span><span class="op">(</span>
  <span class="va">gStats</span>,
  msigGsc <span class="op">=</span> <span class="va">siggs</span>,
  groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="va">g</span><span class="op">]</span>,
  statName <span class="op">=</span> <span class="st">'logFC'</span>,
  topN <span class="op">=</span> <span class="fl">25</span>
<span class="op">)</span> <span class="op">+</span> <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, lty <span class="op">=</span> <span class="fl">2</span>, colour <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="va">p4_g</span> <span class="op">=</span> <span class="fu"><a href="https://DavisLaboratory.github.io/vissE/reference/plotMsigPPI.html">plotMsigPPI</a></span><span class="op">(</span>
  ppidf <span class="op">=</span> <span class="va">ppi</span>,
  msigGsc <span class="op">=</span> <span class="va">siggs</span>,
  groups <span class="op">=</span> <span class="va">grps</span><span class="op">[</span><span class="va">g</span><span class="op">]</span>,
  geneStat <span class="op">=</span> <span class="va">gStats</span>,
  threshFrequency <span class="op">=</span> <span class="fl">0.25</span>, <span class="co">#threshold on gene frequency per cluster</span>
  threshConfidence <span class="op">=</span> <span class="fl">0.2</span>, <span class="co">#threshold on PPI edge confidence</span>
  threshStatistic <span class="op">=</span> <span class="fl">0.5</span>, <span class="co">#threshold on gene-level statistic</span>
  threshUseAbsolute <span class="op">=</span> <span class="cn">TRUE</span> <span class="co">#apply threshold to absolute values</span>
<span class="op">)</span>
<span class="va">p1_g</span> <span class="op">+</span> <span class="va">p2_g</span> <span class="op">+</span> <span class="va">p3_g</span> <span class="op">+</span> <span class="va">p4_g</span> <span class="op">+</span> <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html">plot_layout</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="workflow_singscore_vissE_Chinese_ver_files/figure-html/unnamed-chunk-38-1.png" width="1440"></p>
<p>使用基因集名称计算的词云暗示了一个围绕角化的主题。当使用基因集描述注释类群时，这一点进一步得到强调。这个主题的另一个证据是基因水平的统计图，它显示了角蛋白基因(<em>KRT</em>s)具有高logFC和频率。基因集类群中的基因集紧密聚集，表明它们之间共享许多基因。该类群主要由与PMC42-ET相比在PMC42-LA中上调的基因集组成。正如前文的背景生物学部分所讨论的，富含角蛋白的粘附连接是上皮表型的特征，这与vissE分析提供的可视化结果一致。</p>
</div>
</div>
<div id="总结" class="section level1">
<h1 class="hasAnchor">
<a href="#%E6%80%BB%E7%BB%93" class="anchor"></a>总结</h1>
<p>功能性分析是生物信息学流程的重要组成部分，也是连接生物信息学与生物学的关键环节。此分析流程中提供的工具允许用户使用基因集富集分析来研究生物系统中的分子表型。这些工具和分析流程重点强调了可视化，因为这是与数据交互和从分子数据中阐释有意义的生物学过程的重要因素。</p>
</div>
<div id="使用的包" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#%E4%BD%BF%E7%94%A8%E7%9A%84%E5%8C%85" class="anchor"></a>使用的包</h1>
<p>本次分析流程依赖于Bioconductor项目的3.14中的各种包，它们在R version 4.1.1 (2021-08-10)或更高版本上运行。用于此流程的包的完整列表如下:</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.3 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] igraph_1.2.7                  patchwork_1.1.1              
##  [3] vissE_1.1.7                   msigdb_1.1.5                 
##  [5] singscore_1.14.0              emtdata_1.1.0                
##  [7] GSEABase_1.56.0               graph_1.72.0                 
##  [9] annotate_1.72.0               XML_3.99-0.8                 
## [11] AnnotationDbi_1.56.0          edgeR_3.36.0                 
## [13] limma_3.50.0                  ExperimentHub_2.2.0          
## [15] AnnotationHub_3.2.0           BiocFileCache_2.2.0          
## [17] dbplyr_2.1.1                  SummarizedExperiment_1.24.0  
## [19] Biobase_2.54.0                GenomicRanges_1.46.0         
## [21] GenomeInfoDb_1.30.0           IRanges_2.28.0               
## [23] S4Vectors_0.32.0              BiocGenerics_0.40.0          
## [25] MatrixGenerics_1.6.0          matrixStats_0.61.0           
## [27] ggplot2_3.3.5                 GenesetAnalysisWorkflow_0.9.2
## 
## loaded via a namespace (and not attached):
##   [1] systemfonts_1.0.3             plyr_1.8.6                   
##   [3] splines_4.1.1                 digest_0.6.28                
##   [5] htmltools_0.5.2               viridis_0.6.2                
##   [7] fansi_0.5.0                   magrittr_2.0.1               
##   [9] memoise_2.0.0                 tm_0.7-8                     
##  [11] Biostrings_2.62.0             graphlayouts_0.7.1           
##  [13] textshape_1.7.3               pkgdown_1.6.1                
##  [15] colorspace_2.0-2              blob_1.2.2                   
##  [17] rappdirs_0.3.3                ggrepel_0.9.1                
##  [19] sylly_0.1-6                   textshaping_0.3.6            
##  [21] xfun_0.27                     dplyr_1.0.7                  
##  [23] crayon_1.4.1                  RCurl_1.98-1.5               
##  [25] jsonlite_1.7.2                hexbin_1.28.2                
##  [27] glue_1.4.2                    polyclip_1.10-0              
##  [29] gtable_0.3.0                  zlibbioc_1.40.0              
##  [31] XVector_0.34.0                DelayedArray_0.20.0          
##  [33] scico_1.2.0                   scales_1.1.1                 
##  [35] qdapRegex_0.7.2               DBI_1.1.1                    
##  [37] Rcpp_1.0.7                    viridisLite_0.4.0            
##  [39] xtable_1.8-4                  bit_4.0.4                    
##  [41] textclean_0.9.3               httr_1.4.2                   
##  [43] ggwordcloud_0.5.0             RColorBrewer_1.1-2           
##  [45] ellipsis_0.3.2                pkgconfig_2.0.3              
##  [47] farver_2.1.0                  sass_0.4.0                   
##  [49] locfit_1.5-9.4                utf8_1.2.2                   
##  [51] tidyselect_1.1.1              labeling_0.4.2               
##  [53] rlang_0.4.12                  reshape2_1.4.4               
##  [55] later_1.3.0                   munsell_0.5.0                
##  [57] BiocVersion_3.14.0            tools_4.1.1                  
##  [59] cachem_1.0.6                  generics_0.1.1               
##  [61] RSQLite_2.2.8                 evaluate_0.14                
##  [63] stringr_1.4.0                 fastmap_1.1.0                
##  [65] yaml_2.2.1                    ragg_1.1.3                   
##  [67] textstem_0.1.4                org.Hs.eg.db_3.14.0          
##  [69] knitr_1.36                    bit64_4.0.5                  
##  [71] fs_1.5.0                      tidygraph_1.2.0              
##  [73] purrr_0.3.4                   KEGGREST_1.34.0              
##  [75] ggraph_2.0.5                  koRpus_0.13-8                
##  [77] mime_0.12                     slam_0.1-48                  
##  [79] xml2_1.3.2                    compiler_4.1.1               
##  [81] filelock_1.0.2                curl_4.3.2                   
##  [83] png_0.1-7                     interactiveDisplayBase_1.32.0
##  [85] koRpus.lang.en_0.1-4          syuzhet_1.0.6                
##  [87] tibble_3.1.5                  statmod_1.4.36               
##  [89] tweenr_1.0.2                  bslib_0.3.1                  
##  [91] stringi_1.7.5                 highr_0.9                    
##  [93] desc_1.4.0                    lattice_0.20-45              
##  [95] Matrix_1.3-4                  vctrs_0.3.8                  
##  [97] pillar_1.6.4                  lifecycle_1.0.1              
##  [99] BiocManager_1.30.16           jquerylib_0.1.4              
## [101] data.table_1.14.2             bitops_1.0-7                 
## [103] sylly.en_0.1-3                httpuv_1.6.3                 
## [105] R6_2.5.1                      promises_1.2.0.1             
## [107] KernSmooth_2.23-20            gridExtra_2.3                
## [109] lexicon_1.2.1                 MASS_7.3-54                  
## [111] assertthat_0.2.1              rprojroot_2.0.2              
## [113] withr_2.4.2                   GenomeInfoDbData_1.2.7       
## [115] parallel_4.1.1                grid_4.1.1                   
## [117] prettydoc_0.4.1               tidyr_1.1.4                  
## [119] rmarkdown_2.11                ggforce_0.3.3                
## [121] NLP_0.2-1                     shiny_1.7.1</code></pre>
</div>
<div id="致谢" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#%E8%87%B4%E8%B0%A2" class="anchor"></a>致谢</h1>
<p>我们要感谢Ahmed Mohamed关于改进文本挖掘的想法。</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-R-rmarkdown" class="csl-entry">
Allaire, JJ, Yihui Xie, Jonathan McPherson, Javier Luraschi, Kevin Ushey, Aron Atkins, Hadley Wickham, Joe Cheng, Winston Chang, and Richard Iannone. 2021. <em>Rmarkdown: Dynamic Documents for r</em>. <a href="https://CRAN.R-project.org/package=rmarkdown">https://CRAN.R-project.org/package=rmarkdown</a>.
</div>
<div id="ref-bhuva20" class="csl-entry">
Bhuva, Dharmesh D, Joseph Cursons, and Melissa J Davis. 2020. <span>“Stable Gene Expression for Normalisation and Single-Sample Scoring.”</span> <em>Nucleic Acids Research</em> 48 (19): e113–13.
</div>
<div id="ref-bhuva19" class="csl-entry">
Bhuva, Dharmesh D, Momeneh Foroutan, Yi Xie, Ruqian Lyu, Joseph Cursons, and Melissa J Davis. 2019. <span>“Using Singscore to Predict Mutation Status in Acute Myeloid Leukemia from Transcriptomic Signatures.”</span> <em>F1000Research</em> 8.
</div>
<div id="ref-chen16" class="csl-entry">
Chen, Yunshun, Aaron TL Lun, and Gordon K Smyth. 2016. <span>“From Reads to Genes to Pathways: Differential Expression Analysis of RNA-Seq Experiments Using Rsubread and the edgeR Quasi-Likelihood Pipeline.”</span> <em>F1000Research</em> 5.
</div>
<div id="ref-cursons15" class="csl-entry">
Cursons, Joseph, Karl-Johan Leuchowius, Mark Waltham, Eva Tomaskovic-Crook, Momeneh Foroutan, Cameron P Bracken, Andrew Redfern, et al. 2015. <span>“Stimulus-Dependent Differences in Signalling Regulate Epithelial-Mesenchymal Plasticity and Change the Effects of Drugs in Breast Cancer Cell Lines.”</span> <em>Cell Communication and Signaling</em> 13 (1): 1–21.
</div>
<div id="ref-cursons18" class="csl-entry">
Cursons, Joseph, Katherine A Pillman, Kaitlin G Scheer, Philip A Gregory, Momeneh Foroutan, Soroor Hediyeh-Zadeh, John Toubia, et al. 2018. <span>“Combinatorial Targeting by microRNAs Co-Ordinates Post-Transcriptional Control of EMT.”</span> <em>Cell Systems</em> 7 (1): 77–91.
</div>
<div id="ref-foroutan18" class="csl-entry">
Foroutan, Momeneh, Dharmesh D Bhuva, Ruqian Lyu, Kristy Horan, Joseph Cursons, and Melissa J Davis. 2018. <span>“Single Sample Scoring of Molecular Phenotypes.”</span> <em>BMC Bioinformatics</em> 19 (1): 404.
</div>
<div id="ref-foroutan17" class="csl-entry">
Foroutan, Momeneh, Joseph Cursons, Soroor Hediyeh-Zadeh, Erik W Thompson, and Melissa J Davis. 2017. <span>“A Transcriptional Program for Detecting TGF<span class="math inline">\(\beta\)</span>-Induced EMT in Cancer.”</span> <em>Molecular Cancer Research</em> 15 (5): 619–31.
</div>
<div id="ref-ksiazkiewicz2012" class="csl-entry">
Ksiazkiewicz, M, A Markiewicz, and A J Zaczek. 2012. <span>“Epithelial-Mesenchymal Transition: A Hallmark in Metastasis Formation Linking Circulating Tumor Cells and Cancer Stem Cells.”</span> <em>Pathobiology</em> 79.
</div>
<div id="ref-liao13" class="csl-entry">
Liao, Yang, Gordon K Smyth, and Wei Shi. 2013. <span>“The Subread Aligner: Fast, Accurate and Scalable Read Mapping by Seed-and-Vote.”</span> <em>Nucleic Acids Research</em> 41 (10): e108–8.
</div>
<div id="ref-liao14" class="csl-entry">
———. 2014. <span>“featureCounts: An Efficient General Purpose Program for Assigning Sequence Reads to Genomic Features.”</span> <em>Bioinformatics</em> 30 (7): 923–30.
</div>
<div id="ref-liberzon15" class="csl-entry">
Liberzon, Arthur, Chet Birger, Helga Thorvaldsdóttir, Mahmoud Ghandi, Jill P Mesirov, and Pablo Tamayo. 2015. <span>“The Molecular Signatures Database Hallmark Gene Set Collection.”</span> <em>Cell Systems</em> 1 (6): 417–25.
</div>
<div id="ref-polyak2009" class="csl-entry">
Polyak, Kornelia, and Robert A Weinberg. 2009. <span>“Transitions Between Epithelial and Mesenchymal States: Acquisition of Malignant and Stem Cell Traits.”</span> <em>Nature Reviews Cancer</em> 9.
</div>
<div id="ref-R-prettydoc" class="csl-entry">
Qiu, Yixuan. 2021. <em>Prettydoc: Creating Pretty Documents from r Markdown</em>. <a href="https://github.com/yixuan/prettydoc">https://github.com/yixuan/prettydoc</a>.
</div>
<div id="ref-ribeiro2015" class="csl-entry">
Ribeiro, Ana Sofia, and Joana Paredes. 2015. <span>“P-Cadherin Linking Breast Cancer Stem Cells and Invasion: A Promising Marker to Identify an ’Intermediate/Metastable’ EMT State.”</span> <em>Frontiers in Oncology</em> 4.
</div>
<div id="ref-subramanian05" class="csl-entry">
Subramanian, Aravind, Pablo Tamayo, Vamsi K Mootha, Sayan Mukherjee, Benjamin L Ebert, Michael A Gillette, Amanda Paulovich, et al. 2005. <span>“Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.”</span> <em>Proceedings of the National Academy of Sciences</em> 102 (43): 15545–50.
</div>
<div id="ref-tan14" class="csl-entry">
Tan, Tuan Zea, Qing Hao Miow, Yoshio Miki, Tetsuo Noda, Seiichi Mori, Ruby Yun-Ju Huang, and Jean Paul Thiery. 2014. <span>“Epithelial-Mesenchymal Transition Spectrum Quantification and Its Efficacy in Deciphering Survival and Drug Responses of Cancer Patients.”</span> <em>EMBO Molecular Medicine</em> 6 (10): 1279–93.
</div>
<div id="ref-thiery2009" class="csl-entry">
Thiery, Jean Paul, Herve Acloque, Ruby Y J Huang, and M Angela Nieto. 2009. <span>“Epithelial-Mesenchymal Transitions in Development and Disease.”</span> <em>Cell</em> 139.
</div>
<div id="ref-tomita2000" class="csl-entry">
Tomita, Kyoichi, Adrie van Bokhoven, Geert J L H van Leenders, Emiel T G Ruijter, Cornelius F J Jansen, Marion J G Bussemakers, and Jack A Schalken. 2000. <span>“Cadherin Switching in Human Prostate Cancer Progression.”</span> <em>Cancer Research</em> 60.
</div>
<div id="ref-R-ggplot2" class="csl-entry">
Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen, Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, and Dewey Dunnington. 2021. <em>Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics</em>. <a href="https://CRAN.R-project.org/package=ggplot2">https://CRAN.R-project.org/package=ggplot2</a>.
</div>
<div id="ref-R-knitr" class="csl-entry">
Xie, Yihui. 2021. <em>Knitr: A General-Purpose Package for Dynamic Report Generation in r</em>. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
<div id="ref-yang16" class="csl-entry">
Yang, Zhao, René Algesheimer, and Claudio J Tessone. 2016. <span>“A Comparative Analysis of Community Detection Algorithms on Artificial Networks.”</span> <em>Scientific Reports</em> 6 (1): 1–18.
</div>
<div id="ref-yilmaz2009" class="csl-entry">
Yilmaz, Mahmut, and Gerhard Christofori. 2009. <span>“EMT, the Cytoskeleton, and Cancer Cell Invasion.”</span> <em>Cancer and Metastasis Reviews</em> 28.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Dharmesh D. Bhuva, Chin Wee Tan, Melissa J. Davis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
